diff --git a/components/desktop/pcsc/PCSC-pcsc-1.8.18/src/spy/Makefile.am b/components/desktop/pcsc/PCSC-pcsc-1.8.18/src/spy/Makefile.am
index 3c3daee..be09837 100644
--- a/src/spy/Makefile.am
+++ b/src/spy/Makefile.am
@@ -9,7 +9,7 @@ libpcscspy_la_SOURCES = \
 man_MANS = pcsc-spy.1
 
 pcsc-spy.1: pcsc-spy.pod
-	pod2man \
+	/usr/perl5/bin/pod2man \
 		--center="PC/SC lite" \
 		--release="$(PACKAGE_NAME) $(PACKAGE_VERSION)" \
 		$(srcdir)/pcsc-spy.pod > $@
