diff --git a/lib/core/include/irods_gsi_object.hpp b/lib/core/include/irods_gsi_object.hpp
index 80c0c4e..5869ce0 100644
--- a/lib/core/include/irods_gsi_object.hpp
+++ b/lib/core/include/irods_gsi_object.hpp
@@ -4,6 +4,7 @@
 #include "irods_error.hpp"
 #include "irods_auth_object.hpp"
 
+#pragma clang diagnostic ignored "-Wunknown-pragmas"
 #include <gssapi.h>
 
 namespace irods {
diff --git a/lib/core/include/irods_krb_object.hpp b/lib/core/include/irods_krb_object.hpp
index 0bf1193..5e68cc2 100644
--- a/lib/core/include/irods_krb_object.hpp
+++ b/lib/core/include/irods_krb_object.hpp
@@ -5,6 +5,7 @@
 #include "irods_auth_object.hpp"
 #include "irods_stacktrace.hpp"
 
+#pragma clang diagnostic ignored "-Wunknown-pragmas"
 #include <gssapi.h>
 
 namespace irods {
