diff --git a/server/core/include/rsLog.hpp b/server/core/include/rsLog.hpp
index f1effda..078e904 100644
--- a/server/core/include/rsLog.hpp
+++ b/server/core/include/rsLog.hpp
@@ -13,7 +13,7 @@
 #define RULE_EXEC_LOGFILE   "reLog"
 #define XMSG_SVR_LOGFILE    "xmsgLog"
 
-#define DEF_LOG_DIR "../../var/lib/irods/log"
+#define DEF_LOG_DIR "/var/lib/irods/log"
 #define PROC_LOG_DIR_NAME "proc"
 // =-=-=-=-=-=-=-
 // JMC - backport 4793
