diff --git a/src/libstrongswan/utils/printf_hook/printf_hook_builtin.c b/src/libstrongswan/utils/printf_hook/printf_hook_builtin.c
index af54940..777fe57 100644
--- a/src/libstrongswan/utils/printf_hook/printf_hook_builtin.c
+++ b/src/libstrongswan/utils/printf_hook/printf_hook_builtin.c
@@ -49,6 +49,8 @@
 #include <errno.h>
 #include <math.h>
 
+#define isgreater(x, y)         __builtin_isgreater(x, y)
+
 #define PRINTF_BUF_LEN 8192
 #define ARGS_MAX 3
 
