diff --git a/plugins/background/gsd-background-manager.c b/plugins/background/gsd-background-manager.c
index 4373453..abc85fd 100644
--- a/plugins/background/gsd-background-manager.c
+++ b/plugins/background/gsd-background-manager.c
@@ -237,8 +239,7 @@ draw_background (GsdBackgroundManager *manager,
                 surface = gnome_bg_create_surface (manager->priv->bg,
                                                    root_window,
                                                    gdk_screen_get_width (screen),
-                                                   gdk_screen_get_height (screen),
-                                                   TRUE);
+                                                   gdk_screen_get_height (screen) );
 
                 if (use_crossfade) {
 
