diff -urN gsd.orig/plugins/media-keys/gsd-media-keys-manager.c gsd.new/plugins/media-keys/gsd-media-keys-manager.c
--- gsd.orig/plugins/media-keys/gsd-media-keys-manager.c	2010-10-15 11:27:47.467649253 +0100
+++ gsd.new/plugins/media-keys/gsd-media-keys-manager.c	2010-10-15 11:27:53.368079533 +0100
@@ -661,6 +661,27 @@
         gconf_client_set_bool (client, TOUCHPAD_ENABLED_KEY, !state, NULL);
 }
 
+static void
+do_display_action (GsdMediaKeysManager *manager)
+{
+        gchar *command;
+        gchar *dispswitch;
+
+        if ((dispswitch = g_find_program_in_path ("dispswitch")) == NULL ) {
+                /* Fall back to known location */
+                dispswitch = g_strdup("/usr/X11/bin/dispswitch");
+        }
+
+        command = g_strdup_printf("%s -toggle", dispswitch );
+
+        if ( command != NULL ) {
+                execute (manager, command, FALSE, FALSE);
+        }
+
+        g_free (dispswitch);
+        g_free (command);
+}
+
 #ifdef HAVE_PULSE
 static void
 update_dialog (GsdMediaKeysManager *manager,
