diff --git a/qtmultimedia/cmake/FindFFmpeg.cmake b/qtmultimedia/cmake/FindFFmpeg.cmake
index 7ca637f..6edc01e 100644
--- a/qtmultimedia/cmake/FindFFmpeg.cmake
+++ b/qtmultimedia/cmake/FindFFmpeg.cmake
@@ -315,7 +315,7 @@ foreach (_component ${FFmpeg_FIND_COMPONENTS})
             __ffmpeg_internal_set_dependencies(${_component})
             target_link_libraries(FFmpeg::${_lowerComponent} INTERFACE "${${_component}_LIBRARY_NAME}")
             if (UNIX AND NOT APPLE)
-                target_link_options(FFmpeg::${_lowerComponent} INTERFACE  "-Wl,--exclude-libs=lib${_lowerComponent}")
+                target_link_options(FFmpeg::${_lowerComponent} INTERFACE  "-W")
             endif ()
         endif()
     endif()
