commit 7afdb4853842805a3565553e028619bab2964bdd
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Sat Nov 11 11:12:03 2023 -0500

    Qt6 branch (#1709)
    
    * qt6 wip
    
    * building

diff --git a/librecad/src/actions/rs_actiondrawpolyline.h b/librecad/src/actions/rs_actiondrawpolyline.h
index 851207be..148a618a 100644
--- a/librecad/src/actions/rs_actiondrawpolyline.h
+++ b/librecad/src/actions/rs_actiondrawpolyline.h
@@ -34,7 +34,6 @@ namespace mu {
     class Parser;
 }
 
-class QStringList;
 class RS_EntityContainer;
 class RS_GraphicView;
 class RS_Polyline;
diff --git a/librecad/src/cmd/rs_commands.cpp b/librecad/src/cmd/rs_commands.cpp
index 6ce404d4..6ac93358 100755
--- a/librecad/src/cmd/rs_commands.cpp
+++ b/librecad/src/cmd/rs_commands.cpp
@@ -28,6 +28,7 @@
 #include<vector>
 
 #include <QObject>
+#include <QRegularExpression>
 #include <QTextStream>
 
 #include "rs_commands.h"
@@ -1250,9 +1251,9 @@ void RS_Commands::updateAlias(){
                 if (line.isEmpty() || line.at(0)=='#' ) continue;
                 // Read alias
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
-                QStringList txtList = line.split(QRegExp(R"(\s)"),Qt::SkipEmptyParts);
+                QStringList txtList = line.split(QRegularExpression(R"(\s)"),Qt::SkipEmptyParts);
 #else
-                QStringList txtList = line.split(QRegExp(R"(\s)"),QString::SkipEmptyParts);
+                QStringList txtList = line.split(QRegularExpression(R"(\s)"),QString::SkipEmptyParts);
 #endif
                 if (txtList.size()> 1) {
                     //                    qDebug()<<"reading: "<<txtList.at(0)<<"\t"<< txtList.at(1);
@@ -1362,7 +1363,7 @@ RS2::ActionType RS_Commands::keycodeToAction(const QString& code) const {
     }
 
 
-    //    std::cout<<"regex: "<<qPrintable(c)<<" matches: "<< c.contains(QRegularExpression("^[a-z].*",Qt::CaseInsensitive))<<std::endl;
+//    std::cout<<"regex: "<<qPrintable(c)<<" matches: "<< c.contains(QRegularExpression("^[a-z].*",Qt::CaseInsensitive))<<std::endl;
     //    std::cout<<"RS2::ActionType RS_Commands::keycodeToAction("<<qPrintable(c)<<")"<<std::endl;
 
     auto it = shortCommands.find(c);
@@ -1433,4 +1434,29 @@ QString RS_Commands::msgAvailableCommands() {
     return QObject::tr("Available commands:");
 }
 
+/**
+ * @brief extractCliCal, filter cli calculator math expression
+ * @param cmd, cli string
+ * @return math expression for RS_Math:eval();
+ */
+QString RS_Commands::filterCliCal(const QString& cmd)
+{
+
+    QString str=cmd.trimmed();
+    const QRegularExpression calCmd(R"(^(cal|calculate))");
+    if(!(str.contains(calCmd)
+         || str.startsWith(QObject::tr("cal","command to trigger cli calculator"), Qt::CaseInsensitive)
+         || str.startsWith(QObject::tr("calculate","command to trigger cli calculator"), Qt::CaseInsensitive)
+                           )) {
+        return QString();
+    }
+    int index=str.indexOf(QRegularExpression(R"(\s)"));
+    bool spaceFound=(index>=0);
+    str=str.mid(index);
+    index=str.indexOf(QRegularExpression(R"(\S)"));
+    if(!(spaceFound && index>=0)) return QString();
+    str=str.mid(index);
+    return str;
+}
+
 // EOF
diff --git a/librecad/src/lib/engine/rs_blocklist.cpp b/librecad/src/lib/engine/rs_blocklist.cpp
index 7d70be1b..a8b5c713 100644
--- a/librecad/src/lib/engine/rs_blocklist.cpp
+++ b/librecad/src/lib/engine/rs_blocklist.cpp
@@ -27,7 +27,7 @@
 #include <set>
 #include <iostream>
 #include <QString>
-#include <QRegExp>
+#include <QRegularExpression>
 #include "rs_debug.h"
 #include "rs_blocklist.h"
 #include "rs_block.h"
@@ -241,7 +241,7 @@ QString RS_BlockList::newName(const QString& suggestion) {
 		return suggestion;
 
 	QString name=suggestion;
-	QRegExp const rx(R"(-\d+$)");
+	const QRegularExpression rx(R"(-\d+$)");
 	int index=name.lastIndexOf(rx);
 	int i=-1;
 	if(index>0){
diff --git a/librecad/src/lib/engine/rs_font.cpp b/librecad/src/lib/engine/rs_font.cpp
index 6220b97d..1b8b6671 100644
--- a/librecad/src/lib/engine/rs_font.cpp
+++ b/librecad/src/lib/engine/rs_font.cpp
@@ -27,8 +27,9 @@
 
 
 #include <iostream>
+#include <QRegularExpression>
+#include <QStringEncoder>
 #include <QTextStream>
-#include <QTextCodec>
 
 #include "rs_font.h"
 #include "rs_arc.h"
@@ -182,7 +183,7 @@ void RS_Font::readCXF(QString path) {
             } else if (identifier.toLower()=="name") {
                 names.append(value);
             } else if (identifier.toLower()=="encoding") {
-                ts.setCodec(QTextCodec::codecForName(value.toLatin1()));
+                ts.setEncoding(QStringConverter::encodingForName(value.toLatin1()).value());
                 encoding = value;
             }
         }
@@ -194,11 +195,11 @@ void RS_Font::readCXF(QString path) {
             QChar ch;
 
             // read unicode:
-            QRegExp regexp("[0-9A-Fa-f]{4,4}");
-            regexp.indexIn(line);
-            QString cap = regexp.cap();
-            if (!cap.isNull()) {
-				int uCode = cap.toInt(nullptr, 16);
+            QRegularExpression regexp("[0-9A-Fa-f]{4,4}");
+            QRegularExpressionMatch match=regexp.match(line);
+            if (match.hasMatch()) {
+	        QString cap = match.captured(0);
+		int uCode = cap.toInt(nullptr, 16);
                 ch = QChar(uCode);
             }
 
@@ -321,7 +322,7 @@ void RS_Font::readLFF(QString path) {
             } else if (identifier.toLower()=="license") {
                 fileLicense = value;
             } else if (identifier.toLower()=="encoding") {
-                ts.setCodec(QTextCodec::codecForName(value.toLatin1()));
+                ts.setEncoding(QStringConverter::encodingForName(value.toLatin1()).value());
                 encoding = value;
             } else if (identifier.toLower()=="created") {
                 fileCreate = value;
@@ -335,11 +336,11 @@ void RS_Font::readLFF(QString path) {
             QChar ch;
 
             // read unicode:
-            QRegExp regexp("[0-9A-Fa-f]{1,5}");
-            regexp.indexIn(line);
-            QString cap = regexp.cap();
-            if (!cap.isNull()) {
-				int uCode = cap.toInt(nullptr, 16);
+            QRegularExpression regexp("[0-9A-Fa-f]{1,5}");
+	    QRegularExpressionMatch match=regexp.match(line);
+            if (match.hasMatch()) {
+	        QString cap = match.captured(0);
+		int uCode = cap.toInt(nullptr, 16);
                 ch = QChar(uCode);
             }
             // only unicode allowed
diff --git a/librecad/src/lib/engine/rs_system.cpp b/librecad/src/lib/engine/rs_system.cpp
index 4878bd56..74b56e90 100644
--- a/librecad/src/lib/engine/rs_system.cpp
+++ b/librecad/src/lib/engine/rs_system.cpp
@@ -25,11 +25,12 @@
 **
 **********************************************************************/
 
-#include <QMap>
 #include <QApplication>
-#include <QTextCodec>
-#include <QTranslator>
 #include <QFileInfo>
+#include <QMap>
+#include <QRegularExpression>
+#include <QStringConverter>
+#include <QTranslator>
 #include "rs_settings.h"
 #include "rs_system.h"
 #include "rs.h"
@@ -522,7 +523,7 @@ bool RS_System::createPaths(const QString& directory) {
  */
 QString RS_System::getAppDataDir() {
     QString appData =
-            QStandardPaths::writableLocation( QStandardPaths::DataLocation);
+            QStandardPaths::writableLocation( QStandardPaths::AppDataLocation);
     QDir dir( appData);
     if (!dir.exists()) {
         if (!dir.mkpath( appData))
@@ -645,23 +646,23 @@ QStringList RS_System::getDirectoryList(const QString& _subDirectory) {
     if (subDirectory == "fonts") {
         QString savedFonts = RS_SETTINGS->readEntry( "/Fonts", "");
         RS_DEBUG->print("saved fonts: %s\n", savedFonts.toStdString().c_str());
-        dirList += (RS_SETTINGS->readEntry( "/Fonts", "")).split( QRegExp("[;]"),
+        dirList += (RS_SETTINGS->readEntry( "/Fonts", "")).split( QRegularExpression("[;]"),
                                                                   option);
     }
     else if (subDirectory == "patterns") {
-        dirList += (RS_SETTINGS->readEntry( "/Patterns", "")).split( QRegExp("[;]"),
+        dirList += (RS_SETTINGS->readEntry( "/Patterns", "")).split( QRegularExpression("[;]"),
                                                                   option);
     }
     else if (subDirectory.startsWith( "scripts")) {
-        dirList += (RS_SETTINGS->readEntry( "/Scripts", "")).split( QRegExp("[;]"),
+        dirList += (RS_SETTINGS->readEntry( "/Scripts", "")).split( QRegularExpression("[;]"),
                                                                   option);
     }
     else if (subDirectory.startsWith( "library")) {
-        dirList += (RS_SETTINGS->readEntry( "/Library", "")).split( QRegExp("[;]"),
+        dirList += (RS_SETTINGS->readEntry( "/Library", "")).split( QRegularExpression("[;]"),
                                                                   option);
     }
     else if (subDirectory.startsWith( "qm")) {
-        dirList += (RS_SETTINGS->readEntry( "/Translations", "")).split( QRegExp("[;]"),
+        dirList += (RS_SETTINGS->readEntry( "/Translations", "")).split( QRegularExpression("[;]"),
                                                                   option);
     }
     RS_SETTINGS->endGroup();
@@ -725,7 +726,7 @@ QString RS_System::symbolToLanguage(const QString& symb) {
     QString territory = RS_Locale::countryToString(loc.country());
 #endif
 
-    if (symb.contains( QRegExp( "^en"))) {
+if (symb.contains( QRegularExpression( "^en"))) {
         ret = RS_Locale::languageToString( loc.language());
         if( symb.contains('_') ) {
             ret += " (" + territory + ')';
diff --git a/librecad/src/lib/filters/rs_filterdxf.cpp b/librecad/src/lib/filters/rs_filterdxf.cpp
index aa83cf63..ee8b65e1 100644
--- a/librecad/src/lib/filters/rs_filterdxf.cpp
+++ b/librecad/src/lib/filters/rs_filterdxf.cpp
@@ -145,7 +145,7 @@ bool RS_FilterDXF::fileImport(RS_Graphic& g, const QString& file, RS2::FormatTyp
 QString RS_FilterDXF::getDXFEncoding() {
 
     QString acadver=variables.getString("$ACADVER", "");
-    acadver.replace(QRegExp("[a-zA-Z]"), "");
+    acadver.replace(QRegularExpression("[a-zA-Z]"), "");
     bool ok;
     int version=acadver.toInt(&ok);
 
@@ -3068,28 +3068,28 @@ QString RS_FilterDXF::toNativeString(const char* data, const QString& encoding)
     }
 
     // Line feed:
-    res = res.replace(QRegExp("\\\\P"), "\n");
+    res = res.replace(QRegularExpression("\\\\P"), "\n");
     // Space:
-    res = res.replace(QRegExp("\\\\~"), " ");
+    res = res.replace(QRegularExpression("\\\\~"), " ");
     // diameter:
-    res = res.replace(QRegExp("%%c"), QChar(0x2205));
+    res = res.replace(QRegularExpression("%%c"), QChar(0x2205));
     // degree:
-    res = res.replace(QRegExp("%%d"), QChar(0x00B0));
+    res = res.replace(QRegularExpression("%%d"), QChar(0x00B0));
     // plus/minus
-    res = res.replace(QRegExp("%%p"), QChar(0x00B1));
+    res = res.replace(QRegularExpression("%%p"), QChar(0x00B1));
 
     // Unicode characters:
     QString cap = "";
     int uCode = 0;
     bool ok = false;
     do {
-        QRegExp regexp("\\\\U\\+[0-9A-Fa-f]{4,4}");
+        QRegularExpression regexp("\\\\U\\+[0-9A-Fa-f]{4,4}");
         regexp.indexIn(res);
         cap = regexp.cap();
         if (!cap.isNull()) {
             uCode = cap.right(4).toInt(&ok, 16);
             // workaround for Qt 3.0.x:
-            res.replace(QRegExp("\\\\U\\+" + cap.right(4)), QChar(uCode));
+            res.replace(QRegularExpression("\\\\U\\+" + cap.right(4)), QChar(uCode));
             // for Qt 3.1:
             //res.replace(cap, QChar(uCode));
         }
@@ -3101,13 +3101,13 @@ QString RS_FilterDXF::toNativeString(const char* data, const QString& encoding)
     uCode = 0;
     ok = false;
     do {
-        QRegExp regexp("%%[0-9]{3,3}");
+        QRegularExpression regexp("%%[0-9]{3,3}");
         regexp.indexIn(res);
         cap = regexp.cap();
         if (!cap.isNull()) {
             uCode = cap.right(3).toInt(&ok, 10);
             // workaround for Qt 3.0.x:
-            res.replace(QRegExp("%%" + cap.right(3)), QChar(uCode));
+            res.replace(QRegularExpression("%%" + cap.right(3)), QChar(uCode));
             // for Qt 3.1:
             //res.replace(cap, QChar(uCode));
         }
@@ -3115,7 +3115,7 @@ QString RS_FilterDXF::toNativeString(const char* data, const QString& encoding)
     while (!cap.isNull());
 
     // Ignore font tags:
-    res = res.replace(QRegExp("\\\\f[0-9A-Za-z| ]{0,};"), "");
+    res = res.replace(QRegularExpression("\\\\f[0-9A-Za-z| ]{0,};"), "");
 
     // Ignore {}
     res = res.replace("\\{", "#curly#");
diff --git a/librecad/src/lib/filters/rs_filterdxf1.cpp b/librecad/src/lib/filters/rs_filterdxf1.cpp
index 06f95fc7..1547a889 100644
--- a/librecad/src/lib/filters/rs_filterdxf1.cpp
+++ b/librecad/src/lib/filters/rs_filterdxf1.cpp
@@ -25,6 +25,8 @@
 **********************************************************************/
 
 #include<cstdlib>
+#include<QRegularExpression>
+
 #include "rs_filterdxfrw.h"
 #include "rs_filterdxf1.h"
 
@@ -612,11 +614,11 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     //}
                                     break;
                                 case 10:  // X1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vx1 = dxfLine.toDouble();
                                     break;
                                 case 20:  // Y1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vy1 = dxfLine.toDouble();
                                     break;
                                 case 39:  // Thickness
@@ -664,19 +666,19 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     //}
                                     break;
                                 case 10:  // X1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vx1 = dxfLine.toDouble();
                                     break;
                                 case 20:  // Y1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vy1 = dxfLine.toDouble();
                                     break;
                                 case 11:  // X2
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vx2 = dxfLine.toDouble();
                                     break;
                                 case 21:  // Y2
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vy2 = dxfLine.toDouble();
                                     break;
                                 case 39:  // Thickness
@@ -726,23 +728,23 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     //}
                                     break;
                                 case 10:  // Centre X
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcx = dxfLine.toDouble();
                                     break;
                                 case 20:  // Centre Y
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcy = dxfLine.toDouble();
                                     break;
                                 case 40:  // Radius
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcr = dxfLine.toDouble();
                                     break;
                                 case 50:  // Start Angle
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     va1 = RS_Math::correctAngle(dxfLine.toDouble()/ARAD);
                                     break;
                                 case 51:  // End Angle
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     va2 = RS_Math::correctAngle(dxfLine.toDouble()/ARAD);
                                     break;
                                 case 39:  // Thickness
@@ -791,15 +793,15 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     //}
                                     break;
                                 case 10:  // Centre X
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcx = dxfLine.toDouble();
                                     break;
                                 case 20:  // Centre Y
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcy = dxfLine.toDouble();
                                     break;
                                 case 40:  // Radius
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vcr = dxfLine.toDouble();
                                     break;
                                 case 39:  // Thickness
@@ -1000,15 +1002,15 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     break;
 
                                 case 10:  // X1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vx1 = dxfLine.toDouble();
                                     break;
                                 case 20:  // Y1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vy1 = dxfLine.toDouble();
                                     break;
                                 case 40:  // height
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vheight = dxfLine.toDouble();
                                     /*if(!codeSeven) {
                                         vletterspace = vheight*0.2;
@@ -1016,7 +1018,7 @@ bool RS_FilterDXF1::readFromBuffer() {
                                 }*/
                                     break;
                                 case 50:  // angle
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     vtextAng = dxfLine.toDouble() / ARAD;
                                     break;
                                 case 72:  {// alignment
@@ -1094,7 +1096,7 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     if(dimText==" " || dimText==";;")
                                         dimText="";
 
-                                    //else dimText.replace(QRegExp("%%c"), "¯");
+                                    //else dimText.replace(QRegularExpression("%%c"), "¯");
                                     else
                                         strDecodeDxfString(dimText);
                                     break;
@@ -1111,51 +1113,51 @@ bool RS_FilterDXF1::readFromBuffer() {
                                     //}
                                     break;
                                 case 10:  // line position x
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v10 = dxfLine.toDouble();
                                     break;
                                 case 20:  // line position y
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v20 = dxfLine.toDouble();
                                     break;
                                 case 13:  // X1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v13 = dxfLine.toDouble();
                                     break;
                                 case 23:  // Y1
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v23 = dxfLine.toDouble();
                                     break;
                                 case 14:  // X2
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v14 = dxfLine.toDouble();
                                     break;
                                 case 24:  // Y2
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v24 = dxfLine.toDouble();
                                     break;
                                 case 15:  // X3
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v15 = dxfLine.toDouble();
                                     break;
                                 case 25:  // Y3
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v25 = dxfLine.toDouble();
                                     break;
                                 case 16:  // X4
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v16 = dxfLine.toDouble();
                                     break;
                                 case 26:  // Y4
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v26 = dxfLine.toDouble();
                                     break;
                                 case 40:
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v40 = dxfLine.toDouble();
                                     break;
                                 case 50:
-                                    dxfLine.replace( QRegExp(","), "." );
+                                    dxfLine.replace( QRegularExpression(","), "." );
                                     v50 = dxfLine.toDouble();
                                     break;
                                 case 70:  // Typ
@@ -1442,19 +1444,19 @@ bool RS_FilterDXF1::readFromBuffer() {
                                   //}
                                   break;
                                 case 10:  // Start point/center of boundary line/arc
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v10=dxfLine.toDouble();
                                   break;
                                 case 20:  // Start point/center of boundary line/arc
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v20=dxfLine.toDouble();
                                   break;
                                 case 11:  // End point of boundary line
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v11=dxfLine.toDouble();
                                   break;
                                 case 21:  // End point of boundary line
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v21=dxfLine.toDouble();
                                   if(nextObjectTyp==T_LINE) {
                                     int elnu=graphic->addLine(v10, v20, v11, v21, currentLayerNum, add);
@@ -1462,15 +1464,15 @@ bool RS_FilterDXF1::readFromBuffer() {
                                   }
                                   break;
                                 case 40:  // Radius of boundary entity
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v40=dxfLine.toDouble();
                                   break;
                                 case 50:  // Start angle
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v50=dxfLine.toDouble();
                                   break;
                                 case 51:  // End angle
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   v51=dxfLine.toDouble();
                                   break;
                                 case 73:  // Counterclockwise?
@@ -1489,7 +1491,7 @@ bool RS_FilterDXF1::readFromBuffer() {
                                   }
                                   break;
                                 case 41:  // Scale
-                                  dxfLine.replace( QRegExp(","), "." );
+                                  dxfLine.replace( QRegularExpression(","), "." );
                                   patternScale=dxfLine.toDouble();
                                   break;
                                 case 52:  // Angle
@@ -1856,10 +1858,10 @@ bool RS_FilterDXF1::readFileInBuffer(int _bNum) {
 void RS_FilterDXF1::strDecodeDxfString(QString& str) {
     if (str.isEmpty())
         return;
-    str.replace(QRegExp("%%c"), QChar(0xF8)); // Diameter
-    str.replace(QRegExp("%%d"), QChar(0xB0)); // Degree
-    str.replace(QRegExp("%%p"), QChar(0xB1)); // Plus/minus
-    str.replace(QRegExp("\\\\[pP]"), QChar('\n'));
+    str.replace(QRegularExpression("%%c"), QChar(0xF8)); // Diameter
+    str.replace(QRegularExpression("%%d"), QChar(0xB0)); // Degree
+    str.replace(QRegularExpression("%%p"), QChar(0xB1)); // Plus/minus
+    str.replace(QRegularExpression("\\\\[pP]"), QChar('\n'));
 }
 
 
diff --git a/librecad/src/lib/filters/rs_filterdxfrw.cpp b/librecad/src/lib/filters/rs_filterdxfrw.cpp
index 0b825016..66d78107 100644
--- a/librecad/src/lib/filters/rs_filterdxfrw.cpp
+++ b/librecad/src/lib/filters/rs_filterdxfrw.cpp
@@ -24,8 +24,9 @@
 **********************************************************************/
 
 #include<cstdlib>
+#include <QRegularExpression>
 #include <QStringList>
-#include <QTextCodec>
+#include <QStringConverter>
 
 #include "rs_filterdxfrw.h"
 
@@ -1368,7 +1369,7 @@ void RS_FilterDXFRW::addHeader(const DRW_Header* data){
 		graphic->addVariable("$PDSIZE", LC_DEFAULTS_PDSize, DXF_FORMAT_GC_VarName);
 
     QString acadver = versionStr = graphic->getVariableString("$ACADVER", "");
-    acadver.replace(QRegExp("[a-zA-Z]"), "");
+    acadver.replace(QRegularExpression("[a-zA-Z]"), "");
     bool ok;
     version=acadver.toInt(&ok);
     if (!ok) { version = 1021;}
@@ -3734,11 +3735,11 @@ QString RS_FilterDXFRW::toNativeString(const QString& data) {
                    ) {
                     //found tag, append parsed part
                     res.append(data.mid(j,i-j));
-                    int pos = data.indexOf(0x7D, i+3);//find '}'
+                    qsizetype pos = data.indexOf(QChar(0x7D), i+3);//find '}'
                     if (pos <0) break; //'}' not found
                     QString tmp = data.mid(i+1, pos-i-1);
                     do {
-                        tmp = tmp.remove(0,tmp.indexOf(0x3B, 0)+1 );//remove to ';'
+                        tmp = tmp.remove(0,tmp.indexOf(QChar{0x3B}, 0)+1 );//remove to ';'
                     } while(tmp.startsWith("\\f") || tmp.startsWith("\\H") || tmp.startsWith("\\C"));
                     res.append(tmp);
                     i = j = pos;
@@ -3750,17 +3751,17 @@ QString RS_FilterDXFRW::toNativeString(const QString& data) {
     res.append(data.mid(j));
 
     // Line feed:
-    res = res.replace(QRegExp("\\\\P"), "\n");
+    res = res.replace(QRegularExpression("\\\\P"), "\n");
     // Space:
-    res = res.replace(QRegExp("\\\\~"), " ");
+    res = res.replace(QRegularExpression("\\\\~"), " ");
     // Tab:
-    res = res.replace(QRegExp("\\^I"), "    ");//RLZ: change 4 spaces for \t when mtext have support for tab
+    res = res.replace(QRegularExpression("\\^I"), "    ");//RLZ: change 4 spaces for \t when mtext have support for tab
     // diameter:
-    res = res.replace(QRegExp("%%[cC]"), QChar(0x2300));//RLZ: Empty_set is 0x2205, diameter is 0x2300 need to add in all fonts
+    res = res.replace(QRegularExpression("%%[cC]"), QChar(0x2300));//RLZ: Empty_set is 0x2205, diameter is 0x2300 need to add in all fonts
     // degree:
-    res = res.replace(QRegExp("%%[dD]"), QChar(0x00B0));
+    res = res.replace(QRegularExpression("%%[dD]"), QChar(0x00B0));
     // plus/minus
-    res = res.replace(QRegExp("%%[pP]"), QChar(0x00B1));
+    res = res.replace(QRegularExpression("%%[pP]"), QChar(0x00B1));
 
     return res;
 }
diff --git a/librecad/src/lib/filters/rs_filterjww.cpp b/librecad/src/lib/filters/rs_filterjww.cpp
index a9693ed6..1c5b6d99 100644
--- a/librecad/src/lib/filters/rs_filterjww.cpp
+++ b/librecad/src/lib/filters/rs_filterjww.cpp
@@ -24,7 +24,8 @@
 **
 **********************************************************************/
 
-#include <QTextCodec>
+#include <QRegularExpression>
+#include <QStringConverter>
 #include "rs_filterjww.h"
 
 #include "dl_attributes.h"
@@ -209,9 +210,9 @@ void RS_FilterJWW::addBlock(const DL_BlockData& data) {
                                                                 variables.getString("$DWGCODEPAGE", "ANSI_1252"));
                         // get the codec for Japanese
                         QString blName = data.name.c_str();
-                        QTextCodec *codec = QTextCodec::codecForName(enc.toLatin1());
-                        if(codec)
-                                blName = codec->toUnicode(data.name.c_str());
+                        std::optional<QStringConverter::Encoding> encoding = QStringConverter::encodingForName(enc.toLatin1());
+                        if(encoding)
+                                blName = QStringEncoder{QStringEncoder::Utf8}(QString::fromStdString(data.name));
 //////////////////////////////
                         RS_Block* block =
                                 new RS_Block(graphic,
@@ -487,7 +488,7 @@ void RS_FilterJWW::addMTextChunk(const char* text) {
 QString RS_FilterJWW::getDXFEncoding() {
 
     QString acadver=variables.getString("$ACADVER", "");
-    acadver.replace(QRegExp("[a-zA-Z]"), "");
+    acadver.replace(QRegularExpression("[a-zA-Z]"), "");
     bool ok;
     int version=acadver.toInt(&ok);
 
@@ -2381,7 +2382,7 @@ void RS_FilterJWW::writeEntityContainer(DL_WriterA& dw, RS_EntityContainer* con,
 
         while (true) {
                 tmp = tmp/c;
-                blkName.append((char) tmp %10 + 48);
+                blkName.append(QChar{tmp %10 + 48});
                 c *= 10;
                 if (tmp < 10) {
                         break;
@@ -2484,9 +2485,9 @@ void RS_FilterJWW::setEntityAttributes(RS_Entity* entity,
                                                         variables.getString("$DWGCODEPAGE", "ANSI_1252"));
                 // get the codec for Japanese
                 QString lName = attrib.getLayer().c_str();
-                QTextCodec *codec = QTextCodec::codecForName(enc.toLatin1());
-                if(codec)
-                        lName = codec->toUnicode(attrib.getLayer().c_str());
+                auto encoding = QStringConverter::encodingForName(enc.toLatin1());
+                if(encoding)
+                        lName = QStringEncoder{QStringEncoder::Utf8}(QString::fromStdString(attrib.getLayer()));
 				if (!graphic->findLayer(lName)) {
                         addLayer(DL_LayerData(attrib.getLayer(), 0));
                 }
@@ -3080,34 +3081,33 @@ QString RS_FilterJWW::toNativeString(const char* data, const QString& encoding)
      *	  the string through a textcoder.
      *	--------------------------------------------------------------------- */
     if (!res.contains("\\U+")) {
-        QTextCodec *codec = QTextCodec::codecForName(encoding.toLatin1());
-        if (codec)
-            res = codec->toUnicode(data);
+        auto qEncoding = QStringConverter::encodingForName(encoding.toLatin1());
+        if (qEncoding)
+            res = QStringEncoder{QStringEncoder::Utf8}(data);
     }
 
     // Line feed:
-    res = res.replace(QRegExp("\\\\P"), "\n");
+    res = res.replace(QRegularExpression("\\\\P"), "\n");
     // Space:
-    res = res.replace(QRegExp("\\\\~"), " ");
+    res = res.replace(QRegularExpression("\\\\~"), " ");
     // diameter:
-    res = res.replace(QRegExp("%%c"), QChar(0x2205));
+    res = res.replace(QRegularExpression("%%c"), QChar(0x2205));
     // degree:
-    res = res.replace(QRegExp("%%d"), QChar(0x00B0));
+    res = res.replace(QRegularExpression("%%d"), QChar(0x00B0));
     // plus/minus
-    res = res.replace(QRegExp("%%p"), QChar(0x00B1));
+    res = res.replace(QRegularExpression("%%p"), QChar(0x00B1));
 
     // Unicode characters:
     QString cap = "";
     int uCode = 0;
     bool ok = false;
     do {
-        QRegExp regexp("\\\\U\\+[0-9A-Fa-f]{4,4}");
-        regexp.indexIn(res);
-        cap = regexp.cap();
-        if (!cap.isNull()) {
-            uCode = cap.right(4).toInt(&ok, 16);
+        QRegularExpression regexp("\\\\U\\+[0-9A-Fa-f]{4,4}");
+        QRegularExpressionMatch match=regexp.match(res);
+        if (match.hasMatch()) {
+            uCode = match.captured(0).right(4).toInt(&ok, 16);
             // workaround for Qt 3.0.x:
-            res.replace(QRegExp("\\\\U\\+" + cap.right(4)), QChar(uCode));
+            res.replace(QRegularExpression("\\\\U\\+" + cap.right(4)), QChar(uCode));
             // for Qt 3.1:
             //res.replace(cap, QChar(uCode));
         }
@@ -3119,13 +3119,12 @@ QString RS_FilterJWW::toNativeString(const char* data, const QString& encoding)
 //    uCode = 0;
     ok = false;
     do {
-        QRegExp regexp("%%[0-9]{3,3}");
-        regexp.indexIn(res);
-        cap = regexp.cap();
-        if (!cap.isNull()) {
-            uCode = cap.right(3).toInt(&ok, 10);
+        QRegularExpression regexp("%%[0-9]{3,3}");
+        QRegularExpressionMatch match = regexp.match(res);
+        if (match.hasMatch()) {
+            uCode = match.captured(0).right(3).toInt(&ok, 10);
             // workaround for Qt 3.0.x:
-            res.replace(QRegExp("%%" + cap.right(3)), QChar(uCode));
+            res.replace(QRegularExpression("%%" + cap.right(3)), QChar(uCode));
             // for Qt 3.1:
             //res.replace(cap, QChar(uCode));
         }
@@ -3133,7 +3132,7 @@ QString RS_FilterJWW::toNativeString(const char* data, const QString& encoding)
     while (!cap.isNull());
 
     // Ignore font tags:
-    res = res.replace(QRegExp("\\\\f[0-9A-Za-z| ]{0,};"), "");
+    res = res.replace(QRegularExpression("\\\\f[0-9A-Za-z| ]{0,};"), "");
 
     // Ignore {}
     res = res.replace("\\{", "#curly#");
diff --git a/librecad/src/lib/filters/rs_filterlff.cpp b/librecad/src/lib/filters/rs_filterlff.cpp
index 330bcddc..eff33f63 100644
--- a/librecad/src/lib/filters/rs_filterlff.cpp
+++ b/librecad/src/lib/filters/rs_filterlff.cpp
@@ -135,7 +135,7 @@ bool RS_FilterLFF::fileExport(RS_Graphic& g, const QString& file, RS2::FormatTyp
 
     QFile f(file);
     QTextStream ts(&f);
-    ts.setCodec("UTF-8");
+    ts.setEncoding(QStringConverter::Utf8);
     if (f.open(QIODevice::WriteOnly | QIODevice::Truncate)) {
 
 
diff --git a/librecad/src/lib/generators/lc_xmlwriterqxmlstreamwriter.cpp b/librecad/src/lib/generators/lc_xmlwriterqxmlstreamwriter.cpp
index 30b40147..8012aa14 100644
--- a/librecad/src/lib/generators/lc_xmlwriterqxmlstreamwriter.cpp
+++ b/librecad/src/lib/generators/lc_xmlwriterqxmlstreamwriter.cpp
@@ -29,7 +29,7 @@ LC_XMLWriterQXmlStreamWriter::LC_XMLWriterQXmlStreamWriter():
 	xmlWriter(new QXmlStreamWriter(&xml))
 {
 	xmlWriter->setAutoFormatting(true);
-	xmlWriter->setCodec("UTF-8");
+	//xmlWriter->setEncoding("UTF-8");
 }
 
 LC_XMLWriterQXmlStreamWriter::~LC_XMLWriterQXmlStreamWriter() = default;
diff --git a/librecad/src/lib/gui/rs_eventhandler.cpp b/librecad/src/lib/gui/rs_eventhandler.cpp
index 2ddfe0c6..15b55a62 100644
--- a/librecad/src/lib/gui/rs_eventhandler.cpp
+++ b/librecad/src/lib/gui/rs_eventhandler.cpp
@@ -24,7 +24,7 @@
 **
 **********************************************************************/
 
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QAction>
 #include <QMouseEvent>
 #include "rs_eventhandler.h"
@@ -41,22 +41,22 @@ namespace {
         return action != nullptr && !action->isFinished();
     }
 
-    QString evaluateFraction(QString input, QRegExp rx, int index, int tailI)
+    QString evaluateFraction(QString input, QRegularExpression rx, int index, int tailI)
     {
         QString copy = input;
         QString tail =QString{R"(\)"} + QString::number(tailI);
+	QRegularExpressionMatch match = rx.match(copy);
 
-        int pos = 0;
-        if ((pos = rx.indexIn(copy, pos)) != -1) {
+        if (match.hasMatch()) {
+            qsizetype pos = match.capturedStart();
             LC_ERR<<"Evaluate: "<<copy;
-            LC_ERR<<"pos="<<pos<<", rx.matchedLength()="<<rx.matchedLength();
-            QString formula = ((index != 2) ? rx.cap(2) + "+" : QString{}) + rx.cap(index) + "/" + rx.cap(index + 1);
+            QString formula = ((index != 2) ? match.captured(2) + "+" : QString{}) + match.captured(index) + "/" + match.captured(index + 1);
             LC_ERR<<"formula="<<formula;
             QString value = QString{}.setNum(RS_Math::eval(formula));
             LC_ERR<<"formula="<<formula<<": value="<<value;
             return input.left(pos)
-                    + input.mid(pos, rx.matchedLength()).replace(rx, R"( \1)" + value + tail)
-                    + evaluateFraction(input.right(input.size() - pos - rx.matchedLength()), rx, index, tailI);
+                    + input.mid(pos, match.capturedLength()).replace(rx, R"( \1)" + value + tail)
+                    + evaluateFraction(input.right(input.size() - pos - match.capturedLength()), rx, index, tailI);
         }
         return input;
     }
@@ -68,19 +68,20 @@ namespace {
     */
     QString updateForFraction(QString input) {
         // support fraction at the end: (1'1/2) => (1 1/2')
-        QRegExp rx{R"((\D*)([\d]+)\s*(['"])([\d]+)/([\d]+)\s*$)"};
-        int pos = 0;
-        if ((pos = rx.indexIn(input, pos)) != -1) {
-            input = input.left(pos) + rx.cap(1) + rx.cap(2) + " " + rx.cap(4) + "/" + rx.cap(5) + rx.cap(3);
+        QRegularExpression rx{R"((\D*)([\d]+)\s*(['"])([\d]+)/([\d]+)\s*$)"};
+        QRegularExpressionMatch match = rx.match(input);
+        if (match.hasMatch()) {
+	    qsizetype pos = match.capturedStart();
+            input = input.left(pos) + match.captured(1) + match.captured(2) + " " + match.captured(4) + "/" + match.captured(5) + match.captured(3);
         }
-        std::vector<std::tuple<QRegExp, int, int>> regexps{{
-                {QRegExp{R"((\D*)([\d]+)\s+([\d]+)/([\d]+)\s*([\D$]))"}, 3, 5},
-                {QRegExp{R"((\D*)([\d]+)\s+([\d]+)/([\d]+)\s*(['"]))"}, 3, 5},
-                {QRegExp{R"((\D*)\s*([\d]+)/([\d]+)\s*([\D$]))"}, 2, 4},
+        std::vector<std::tuple<QRegularExpression, int, int>> regexps{{
+                {QRegularExpression{R"((\D*)([\d]+)\s+([\d]+)/([\d]+)\s*([\D$]))"}, 3, 5},
+                {QRegularExpression{R"((\D*)([\d]+)\s+([\d]+)/([\d]+)\s*(['"]))"}, 3, 5},
+                {QRegularExpression{R"((\D*)\s*([\d]+)/([\d]+)\s*([\D$]))"}, 2, 4},
             }};
         LC_LOG<<"input="<<input;
         for(auto& [rx, index, tailI] : regexps)
-            input = evaluateFraction(input, rx, index, tailI).replace(QRegExp(R"(\s+)"), QString{});
+            input = evaluateFraction(input, rx, index, tailI).replace(QRegularExpression(R"(\s+)"), QString{});
         LC_LOG<<"eval: "<<input;
         return input;
     }
diff --git a/librecad/src/lib/gui/rs_graphicview.cpp b/librecad/src/lib/gui/rs_graphicview.cpp
index 9f5f307a..6fba408a 100644
--- a/librecad/src/lib/gui/rs_graphicview.cpp
+++ b/librecad/src/lib/gui/rs_graphicview.cpp
@@ -29,7 +29,6 @@
 #include<cmath>
 
 #include <QApplication>
-#include <QDesktopWidget>
 #include <QAction>
 #include <QMouseEvent>
 #include <QtAlgorithms>
@@ -918,7 +917,7 @@ void RS_GraphicView::drawLayer1(RS_Painter *painter) {
 	if (!isPrintPreview()) {
 
 		//increase grid point size on for DPI>96
-		int dpiX = qApp->desktop()->logicalDpiX();
+		auto dpiX = int(qApp->screens().front()->logicalDotsPerInch());
         const bool isHiDpi = dpiX > 96;
 		//        DEBUG_HEADER
 		//        RS_DEBUG->print(RS_Debug::D_ERROR, "dpiX=%d\n",dpiX);
diff --git a/librecad/src/lib/math/rs_math.cpp b/librecad/src/lib/math/rs_math.cpp
index 29ba7e55..337dd2f8 100644
--- a/librecad/src/lib/math/rs_math.cpp
+++ b/librecad/src/lib/math/rs_math.cpp
@@ -413,8 +413,8 @@ QString RS_Math::doubleToString(double value, int prec) {
 
     if(valStr.contains('.')) {
         // Remove tailing point and zeros:
-//        valStr.replace(QRegExp("0*$"), "");
-//        valStr.replace(QRegExp(R"(\.$)"), "");
+//        valStr.replace(QRegularExpression("0*$"), "");
+//        valStr.replace(QRegularExpression(R"(\.$)"), "");
 //        while (valStr.at(valStr.length()-1)=='0') {
 //            valStr.truncate(valStr.length()-1);
 //        }
diff --git a/librecad/src/lib/printing/lc_printing.cpp b/librecad/src/lib/printing/lc_printing.cpp
index e90ca672..89786227 100644
--- a/librecad/src/lib/printing/lc_printing.cpp
+++ b/librecad/src/lib/printing/lc_printing.cpp
@@ -29,36 +29,36 @@
 namespace {
 
 // supported paper formats should be added here
-const std::map<RS2::PaperFormat, QPrinter::PageSize> paperToPage = {
+const std::map<RS2::PaperFormat, QPageSize::PageSizeId> paperToPage = {
     {
-            {RS2::A0, QPrinter::A0},
-            {RS2::A1, QPrinter::A1},
-            {RS2::A2, QPrinter::A2},
-            {RS2::A3, QPrinter::A3},
-            {RS2::A4, QPrinter::A4},
+            {RS2::A0, QPageSize::A0},
+            {RS2::A1, QPageSize::A1},
+            {RS2::A2, QPageSize::A2},
+            {RS2::A3, QPageSize::A3},
+            {RS2::A4, QPageSize::A4},
 
                 /* Removed ISO "B" and "C" series,  C5E,  Comm10E,  DLE,  (envelope sizes) */
 
-            {RS2::Letter, QPrinter::Letter},
-            {RS2::Legal,  QPrinter::Legal},
-            {RS2::Tabloid, QPrinter::Tabloid},
+            {RS2::Letter, QPageSize::Letter},
+            {RS2::Legal,  QPageSize::Legal},
+            {RS2::Tabloid, QPageSize::Tabloid},
 
-                //case RS2::Ansi_A, QPrinter::AnsiA},
-                //case RS2::Ansi_B, QPrinter::AnsiB},
-            {RS2::Ansi_C, QPrinter::AnsiC},
-            {RS2::Ansi_D, QPrinter::AnsiD},
-            {RS2::Ansi_E, QPrinter::AnsiE},
+                //case RS2::Ansi_A, QPageSize::AnsiA},
+                //case RS2::Ansi_B, QPageSize::AnsiB},
+            {RS2::Ansi_C, QPageSize::AnsiC},
+            {RS2::Ansi_D, QPageSize::AnsiD},
+            {RS2::Ansi_E, QPageSize::AnsiE},
 
-            {RS2::Arch_A, QPrinter::ArchA},
-            {RS2::Arch_B, QPrinter::ArchB},
-            {RS2::Arch_C, QPrinter::ArchC},
-            {RS2::Arch_D, QPrinter::ArchD},
-            {RS2::Arch_E, QPrinter::ArchE},
+            {RS2::Arch_A, QPageSize::ArchA},
+            {RS2::Arch_B, QPageSize::ArchB},
+            {RS2::Arch_C, QPageSize::ArchC},
+            {RS2::Arch_D, QPageSize::ArchD},
+            {RS2::Arch_E, QPageSize::ArchE},
     }
 };
 }
 
-QPrinter::PageSize LC_Printing::rsToQtPaperFormat(RS2::PaperFormat paper)
+QPageSize::PageSizeId LC_Printing::rsToQtPaperFormat(RS2::PaperFormat paper)
 {
-    return (paperToPage.count(paper) == 1) ? paperToPage.at(paper) : QPrinter::Custom;
+    return (paperToPage.count(paper) == 1) ? paperToPage.at(paper) : QPageSize::Custom;
 }
diff --git a/librecad/src/lib/printing/lc_printing.h b/librecad/src/lib/printing/lc_printing.h
index 54511956..81fb86a7 100644
--- a/librecad/src/lib/printing/lc_printing.h
+++ b/librecad/src/lib/printing/lc_printing.h
@@ -1,13 +1,13 @@
 #ifndef LC_PRINTING_H
 #define LC_PRINTING_H
 
-#include <QPrinter>
+#include <QPageSize>
 
 #include "rs.h"
 
 namespace LC_Printing
 {
-    QPrinter::PageSize rsToQtPaperFormat(RS2::PaperFormat f);
+    QPageSize::PageSizeId rsToQtPaperFormat(RS2::PaperFormat f);
 }
 
 #endif // LC_PRINTING_H
diff --git a/librecad/src/main/console_dxf2pdf/pdf_print_loop.cpp b/librecad/src/main/console_dxf2pdf/pdf_print_loop.cpp
index 529c91de..215868ef 100644
--- a/librecad/src/main/console_dxf2pdf/pdf_print_loop.cpp
+++ b/librecad/src/main/console_dxf2pdf/pdf_print_loop.cpp
@@ -39,40 +39,6 @@ static void touchGraphic(RS_Graphic*, PdfPrintParams&);
 static void setupPrinterAndPaper(RS_Graphic*, QPrinter&, PdfPrintParams&);
 static void drawPage(RS_Graphic*, QPrinter&, RS_PainterQt&);
 
-namespace {
-// Convert QPrinter::PageSize to QPageSize
-QPageSize toPageSize(QPrinter::PageSize pageSize)
-{
-    static std::map<QPrinter::PageSize, QPageSize::PageSizeId> lookUp = {
-        {QPrinter::A0, QPageSize::A0},
-        {QPrinter::A1, QPageSize::A1},
-        {QPrinter::A2, QPageSize::A2},
-        {QPrinter::A3, QPageSize::A3},
-        {QPrinter::A4, QPageSize::A4},
-
-        /* Removed ISO "B" and "C" series,  C5E,  Comm10E,  DLE,  (envelope sizes) */
-
-        {QPrinter::Letter, QPageSize::Letter},
-        {QPrinter::Legal, QPageSize::Legal},
-        {QPrinter::Tabloid, QPageSize::Tabloid},
-
-        //case RS2::Ansi_A, {QPrinter::AnsiA, QPageSize::AnsiA},
-        //case RS2::Ansi_B, {QPrinter::AnsiB, QPageSize::AnsiB},
-        {QPrinter::AnsiC, QPageSize::AnsiC},
-        {QPrinter::AnsiD, QPageSize::AnsiD},
-        {QPrinter::AnsiE, QPageSize::AnsiE},
-
-        {QPrinter::ArchA, QPageSize::ArchA},
-        {QPrinter::ArchB, QPageSize::ArchB},
-        {QPrinter::ArchC, QPageSize::ArchC},
-        {QPrinter::ArchD, QPageSize::ArchD},
-        {QPrinter::ArchE, QPageSize::ArchE}};
-    QPageSize::PageSizeId id = lookUp.count(pageSize) == 1 ?
-                                   lookUp.at(pageSize) : QPageSize::Custom;
-    return QPageSize{id};
-}
-}
-
 void PdfPrintLoop::run()
 {
     if (params.outFile.isEmpty()) {
@@ -133,7 +99,7 @@ void PdfPrintLoop::printManyDxfToOnePdf() {
         RS_Document* doc;
         RS_Graphic* graphic;
         QString dxfFile;
-        QPrinter::PageSize paperSize;
+        QPageSize::PageSizeId paperSize;
     };
 
     if (!params.outDir.isEmpty()) {
@@ -251,9 +217,9 @@ static void setupPrinterAndPaper(RS_Graphic* graphic, QPrinter& printer,
     bool landscape = false;
 
     RS2::PaperFormat pf = graphic->getPaperFormat(&landscape);
-    QPrinter::PageSize paperSize = LC_Printing::rsToQtPaperFormat(pf);
+    QPageSize::PageSizeId paperSize = LC_Printing::rsToQtPaperFormat(pf);
 
-    if (paperSize == QPrinter::Custom){
+    if (paperSize == QPageSize::Custom){
         RS_Vector r = graphic->getPaperSize();
         RS_Vector s = RS_Units::convert(r, graphic->getUnit(),
             RS2::Millimeter);
@@ -265,11 +231,7 @@ static void setupPrinterAndPaper(RS_Graphic* graphic, QPrinter& printer,
         printer.setPaperSize(QSizeF{s.x,s.y}, QPrinter::Millimeter);
 #endif
     } else {
-#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
-        printer.setPageSize(toPageSize(paperSize));
-#else
-        printer.setPaperSize(paperSize);
-#endif
+        printer.setPageSize(paperSize);
     }
 
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
diff --git a/librecad/src/main/lc_application.cpp b/librecad/src/main/lc_application.cpp
index 96b2c648..74142d48 100644
--- a/librecad/src/main/lc_application.cpp
+++ b/librecad/src/main/lc_application.cpp
@@ -24,9 +24,7 @@ along with this program; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 **********************************************************************/
 
-#include <QStringList>
 #include <QFileOpenEvent>
-
 #include "lc_application.h"
 
 LC_Application::LC_Application(int &argc, char **argv)
diff --git a/librecad/src/main/lc_application.h b/librecad/src/main/lc_application.h
index 720334ed..d0701be3 100644
--- a/librecad/src/main/lc_application.h
+++ b/librecad/src/main/lc_application.h
@@ -29,8 +29,6 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 #include <QApplication>
 
-class QStringList;
-
 class LC_Application : public QApplication
 {
     Q_OBJECT
diff --git a/librecad/src/main/main.cpp b/librecad/src/main/main.cpp
index 236d77d6..9a39601d 100644
--- a/librecad/src/main/main.cpp
+++ b/librecad/src/main/main.cpp
@@ -145,14 +145,14 @@ int main(int argc, char** argv)
             // to control the level of debugging output use --debug with level 0-6, e.g. --debug3
             // for a list of debug levels use --debug?
             // if no level follows, the debugging level is set
-            argstr.remove(QRegExp("^"+lpDebugSwitch0));
-            argstr.remove(QRegExp("^"+lpDebugSwitch1));
+            argstr.remove(QRegularExpression("^"+lpDebugSwitch0));
+            argstr.remove(QRegularExpression("^"+lpDebugSwitch1));
             char level;
             if(argstr.size()==0)
             {
                 if(i+1<argc)
                 {
-                    if(QRegExp("\\d*").exactMatch(argv[i+1]))
+                    if(QRegularExpression(R"(\d*)").match(argv[i+1]).hasMatch())
                     {
                         ++i;
                         qDebug()<<"reading "<<argv[i]<<" as debugging level";
diff --git a/librecad/src/main/qc_applicationwindow.cpp b/librecad/src/main/qc_applicationwindow.cpp
index 21129e3a..5fb0442d 100644
--- a/librecad/src/main/qc_applicationwindow.cpp
+++ b/librecad/src/main/qc_applicationwindow.cpp
@@ -42,7 +42,7 @@
 #include <QPluginLoader>
 #include <QPrinter>
 #include <QPrintDialog>
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QSplitter>
 #include <QStatusBar>
 #include <QStyleFactory>
@@ -2157,7 +2158,7 @@ void QC_ApplicationWindow::slotFileExport() {
             QString format = "";
             int i = filter.indexOf("(*.");
             if (i!=-1) {
-                int i2 = filter.indexOf(QRegExp("[) ]"), i);
+                int i2 = filter.indexOf(QRegularExpression("[) ]"), i);
                 format = filter.mid(i+3, i2-(i+3));
                 format = format.toUpper();
             }
@@ -2419,9 +2419,9 @@ void QC_ApplicationWindow::slotFilePrint(bool printPDF) {
 
     bool landscape = false;
     RS2::PaperFormat pf = graphic->getPaperFormat(&landscape);
-    QPrinter::PageSize paperSizeName = LC_Printing::rsToQtPaperFormat(pf);
+    QPageSize::PageSizeId paperSizeName = LC_Printing::rsToQtPaperFormat(pf);
     RS_Vector paperSize = graphic->getPaperSize();
-    if(paperSizeName==QPrinter::Custom){
+    if(paperSizeName==QPageSize::Custom){
         RS_Vector s=RS_Units::convert(paperSize, graphic->getUnit(),RS2::Millimeter);
         if(landscape) s=s.flipXY();
         printer.setPageSize(QPageSize{QSizeF(s.x,s.y), QPageSize::Millimeter});
diff --git a/librecad/src/ui/forms/qg_dlgmtext.cpp b/librecad/src/ui/forms/qg_dlgmtext.cpp
index 8ce9b219..804cbbde 100644
--- a/librecad/src/ui/forms/qg_dlgmtext.cpp
+++ b/librecad/src/ui/forms/qg_dlgmtext.cpp
@@ -25,7 +25,6 @@
 **********************************************************************/
 
 #include <vector>
-#include <QTextCodec>
 #include <QTextStream>
 #include <QFileDialog>
 #include <QPalette>
diff --git a/librecad/src/ui/forms/qg_dlgoptionsdrawing.h b/librecad/src/ui/forms/qg_dlgoptionsdrawing.h
index 91bcba28..e4550b2f 100644
--- a/librecad/src/ui/forms/qg_dlgoptionsdrawing.h
+++ b/librecad/src/ui/forms/qg_dlgoptionsdrawing.h
@@ -30,7 +30,6 @@
 #include <QGraphicsScene>
 #include "ui_qg_dlgoptionsdrawing.h"
 
-class QStringList;
 
 class RS_Graphic;
 class RS_Vector;
diff --git a/librecad/src/ui/forms/qg_dlgtext.cpp b/librecad/src/ui/forms/qg_dlgtext.cpp
index 2fa8934d..9fe49d20 100644
--- a/librecad/src/ui/forms/qg_dlgtext.cpp
+++ b/librecad/src/ui/forms/qg_dlgtext.cpp
@@ -25,7 +25,6 @@
 **********************************************************************/
 #include "qg_dlgtext.h"
 
-#include <QTextCodec>
 #include <QTextStream>
 #include <QFileDialog>
 #include "rs_system.h"
diff --git a/librecad/src/ui/forms/qg_snaptoolbar.cpp b/librecad/src/ui/forms/qg_snaptoolbar.cpp
index bc6eb7ef..8811baeb 100644
--- a/librecad/src/ui/forms/qg_snaptoolbar.cpp
+++ b/librecad/src/ui/forms/qg_snaptoolbar.cpp
@@ -31,8 +31,18 @@
 #include "qg_actionhandler.h"
 #include "lc_actiongroupmanager.h"
 
+namespace {
+QAction* createAction(QWidget* parent, QString img, const QString& text, QActionGroup* group)
+{
+    QAction* action = new QAction(QIcon{img}, text, parent);
+    action->setActionGroup(group);
+    action->setCheckable(true);
+    return action;
+}
+
+}
 /*
- *  Constructs a QG_CadToolBarSnap as a child of 'parent', with the
+ *  Constructs a QG_Cadhttps://apps.e-signlive.ca/error?error=eyJjb2RlIjo0MDEsIm1lc3NhZ2VLZXkiOiJlcnJvci51bmF1dGhvcmlzZWQuc2Vzc2lvbkV4cGlyZWQiLCJ0ZWNobmljYWwiOiJTZXNzaW9uIGlzIGV4cGlyZWQifQ%3D%3DToolBarSnap as a child of 'parent', with the
  *  name 'name' and widget flags set to 'f'.
  */
 QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionGroupManager* agm)
@@ -40,10 +50,8 @@ QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionG
     , actionHandler(ah)
 {
 
-    auto action = new QAction(tr("Exclusive Snap Mode"), agm->snap_extras);
+    auto action = createAction(this, ":/icons/exclusive.svg", tr("Exclusive Snap Mode"), agm->snap_extras);
     action->setShortcut(QKeySequence(Qt::ALT | Qt::Key_X));
-    action->setIcon(QIcon(":/icons/exclusive.svg"));
-    action->setCheckable(true);
     connect(action, SIGNAL(toggled(bool)),
             agm, SLOT(toggleExclusiveSnapMode(bool)));
     action->setObjectName("ExclusiveSnapMode");
@@ -60,37 +68,37 @@ QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionG
     snapFree->setObjectName("SnapFree");
     connect(snapFree, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapFree);
-    snapGrid = new QAction(QIcon(":/icons/snap_grid.svg"), tr("Snap on grid"), agm->snap);
+    snapGrid = createAction(this, ":/icons/snap_grid.svg", tr("Snap on grid"), agm->snap);
     snapGrid->setObjectName("SnapGrid");
     snapGrid->setCheckable(true);
     connect(snapGrid, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapGrid);
-    snapEnd = new QAction(QIcon(":/icons/snap_endpoints.svg"), tr("Snap on Endpoints"), agm->snap);
+    snapEnd = createAction(this, ":/icons/snap_endpoints.svg", tr("Snap on Endpoints"), agm->snap);
     snapEnd->setObjectName("SnapEnd");
     snapEnd->setCheckable(true);
     connect(snapEnd, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapEnd);
-    snapOnEntity = new QAction(QIcon(":/icons/snap_entity.svg"), tr("Snap on Entity"), agm->snap);
+    snapOnEntity = createAction(this, ":/icons/snap_entity.svg", tr("Snap on Entity"), agm->snap);
     snapOnEntity->setObjectName("SnapEntity");
     snapOnEntity->setCheckable(true);
     connect(snapOnEntity, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapOnEntity);
-    snapCenter = new QAction(QIcon(":/icons/snap_center.svg"), tr("Snap Center"), agm->snap);
+    snapCenter = createAction(this, ":/icons/snap_center.svg", tr("Snap Center"), agm->snap);
     snapCenter->setObjectName("SnapCenter");
     snapCenter->setCheckable(true);
     connect(snapCenter, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapCenter);
-    snapMiddle = new QAction(QIcon(":/icons/snap_middle.svg"), tr("Snap Middle"), agm->snap);
+    snapMiddle = createAction(this, ":/icons/snap_middle.svg", tr("Snap Middle"), agm->snap);
     snapMiddle->setObjectName("SnapMiddle");
     snapMiddle->setCheckable(true);
     connect(snapMiddle, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapMiddle);
-    snapDistance = new QAction(QIcon(":/icons/snap_distance.svg"), tr("Snap Distance"), agm->snap);
+    snapDistance = createAction(this, ":/icons/snap_distance.svg", tr("Snap Distance"), agm->snap);
     snapDistance->setObjectName("SnapDistance");
     snapDistance ->setCheckable(true);
     connect(snapDistance, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(snapDistance);
-    snapIntersection = new QAction(QIcon(":/icons/snap_intersection.svg"), tr("Snap Intersection"), agm->snap);
+    snapIntersection = createAction(this, ":/icons/snap_intersection.svg", tr("Snap Intersection"), agm->snap);
     snapIntersection->setObjectName("SnapIntersection");
     snapIntersection->setCheckable(true);
     connect(snapIntersection, SIGNAL(triggered()), this, SLOT(actionTriggered()));
@@ -98,20 +106,20 @@ QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionG
 
     this->addSeparator();
 
-    restrictHorizontal = new QAction(QIcon(":/icons/restr_hor.svg"),
+    restrictHorizontal = createAction(this, ":/icons/restr_hor.svg",
                                      tr("Restrict Horizontal"), agm->restriction);
     restrictHorizontal->setObjectName("RestrictHorizontal");
     restrictHorizontal->setCheckable(true);
     connect(restrictHorizontal, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(restrictHorizontal);
-    restrictVertical = new QAction(QIcon(":/icons/restr_ver.svg"),
+    restrictVertical = createAction(this, ":/icons/restr_ver.svg",
                                    tr("Restrict Vertical"), agm->restriction);
     restrictVertical->setObjectName("RestrictVertical");
     restrictVertical->setCheckable(true);
     connect(restrictVertical, SIGNAL(triggered()), this, SLOT(actionTriggered()));
     this->addAction(restrictVertical);
 
-    restrictOrthogonal = new QAction(QIcon(":/icons/restr_ortho.svg"),
+    restrictOrthogonal = createAction(this, ":/icons/restr_ortho.svg",
                                    tr("Restrict Orthogonal"), agm->restriction);
     restrictOrthogonal->setObjectName("RestrictOrthogonal");
     restrictOrthogonal->setCheckable(true);
@@ -119,7 +127,7 @@ QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionG
             SLOT(slotRestrictOrthogonal(bool)));
 	this->addAction(restrictOrthogonal);
 
-    restrictNothing = new QAction(QIcon(":/extui/restrictnothing.png"),
+    restrictNothing = createAction(this, ":/extui/restrictnothing.png",
                                    tr("Restrict Nothing"), agm->restriction);
     restrictNothing->setObjectName("RestrictNothing");
     restrictNothing->setCheckable(true);
@@ -127,13 +135,13 @@ QG_SnapToolBar::QG_SnapToolBar(QWidget* parent, QG_ActionHandler* ah, LC_ActionG
             SLOT(slotRestrictNothing(bool)));
 
     this->addSeparator();
-    bRelZero = new QAction(QIcon(":/icons/set_rel_zero.svg"), tr("Set relative zero position"), agm->other);
+    bRelZero = createAction(this, ":/icons/set_rel_zero.svg", tr("Set relative zero position"), agm->other);
     bRelZero->setObjectName("SetRelativeZero");
     bRelZero->setCheckable(false);
     connect(bRelZero, SIGNAL(triggered()), actionHandler, SLOT(slotSetRelativeZero()));
     //connect(bRelZero, SIGNAL(triggered()), this, SLOT(slotSetRelativeZero()));
     this->addAction(bRelZero);
-    bLockRelZero = new QAction(QIcon(":/icons/lock_rel_zero.svg"), tr("Lock relative zero position"), agm->other);
+    bLockRelZero = createAction(this, ":/icons/lock_rel_zero.svg", tr("Lock relative zero position"), agm->other);
     bLockRelZero->setObjectName("LockRelativeZero");
     bLockRelZero->setCheckable(true);
     connect(bLockRelZero, SIGNAL(toggled(bool)),actionHandler, SLOT(slotLockRelativeZero(bool)));
diff --git a/librecad/src/ui/lc_actionfactory.cpp b/librecad/src/ui/lc_actionfactory.cpp
index c60cd4e9..af6ce12d 100644
--- a/librecad/src/ui/lc_actionfactory.cpp
+++ b/librecad/src/ui/lc_actionfactory.cpp
@@ -1588,7 +1588,7 @@ void LC_ActionFactory::commonActions(QMap<QString, QAction*>& a_map, LC_ActionGr
 
 	action = new QAction(tr("Save A&ll..."), agm->file);
 	action->setIcon(QIcon(":/icons/save_all.svg"));
-	QKeySequence shortcut = QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_S);
+	QKeySequence shortcut = QKeySequence::SaveAs; //(Qt::CTRL + Qt::SHIFT + Qt::Key_S);
 	// only define this shortcut for platforms not already using it for save as
 	if (shortcut != QKeySequence::SaveAs)
 		action->setShortcut(shortcut);
diff --git a/librecad/src/ui/lc_penwizard.cpp b/librecad/src/ui/lc_penwizard.cpp
index 4c1afb81..1a475b48 100644
--- a/librecad/src/ui/lc_penwizard.cpp
+++ b/librecad/src/ui/lc_penwizard.cpp
@@ -43,7 +43,7 @@ LC_PenWizard::LC_PenWizard(const QString& title, QWidget* parent)
     auto layout = new QVBoxLayout;
     frame->setLayout(layout);
 
-    layout->setMargin(0);
+    layout->setContentsMargins(QMargins{});
     layout->addWidget(color_wiz);
 
     setWidget(frame);
diff --git a/librecad/src/ui/qg_blockwidget.cpp b/librecad/src/ui/qg_blockwidget.cpp
index de9dd262..80936460 100644
--- a/librecad/src/ui/qg_blockwidget.cpp
+++ b/librecad/src/ui/qg_blockwidget.cpp
@@ -391,29 +391,29 @@ void QG_BlockWidget::contextMenuEvent(QContextMenuEvent *e) {
     caption->setPalette(palette);
     caption->setAlignment( Qt::AlignCenter );
     // Actions for all blocks:
-    contextMenu->addAction( tr("&Defreeze all Blocks"), actionHandler,
-                             SLOT(slotBlocksDefreezeAll()), 0);
-    contextMenu->addAction( tr("&Freeze all Blocks"), actionHandler,
-                             SLOT(slotBlocksFreezeAll()), 0);
+    contextMenu->addAction( tr("&Defreeze all Blocks"), 0, actionHandler,
+                             SLOT(slotBlocksDefreezeAll()));
+    contextMenu->addAction( tr("&Freeze all Blocks"), 0, actionHandler,
+                             SLOT(slotBlocksFreezeAll()));
     contextMenu->addSeparator();
     // Actions for selected blocks or,
     // if nothing is selected, for active block:
-    contextMenu->addAction( tr("&Toggle Visibility"), actionHandler,
-                             SLOT(slotBlocksToggleView()), 0);
-    contextMenu->addAction( tr("&Remove Block"), actionHandler,
-                             SLOT(slotBlocksRemove()), 0);
+    contextMenu->addAction( tr("&Toggle Visibility"), 0, actionHandler,
+                             SLOT(slotBlocksToggleView()));
+    contextMenu->addAction( tr("&Remove Block"), 0, actionHandler,
+                             SLOT(slotBlocksRemove()));
     contextMenu->addSeparator();
     // Single block actions:
-    contextMenu->addAction( tr("&Add Block"), actionHandler,
-                             SLOT(slotBlocksAdd()), 0);
-    contextMenu->addAction( tr("&Rename Block"), actionHandler,
-                             SLOT(slotBlocksAttributes()), 0);
-    contextMenu->addAction( tr("&Edit Block"), actionHandler,
-                             SLOT(slotBlocksEdit()), 0);
-    contextMenu->addAction( tr("&Insert Block"), actionHandler,
-                             SLOT(slotBlocksInsert()), 0);
-    contextMenu->addAction( tr("&Create New Block"), actionHandler,
-                             SLOT(slotBlocksCreate()), 0);
+    contextMenu->addAction( tr("&Add Block"), 0, actionHandler,
+                             SLOT(slotBlocksAdd()));
+    contextMenu->addAction( tr("&Rename Block"), 0, actionHandler,
+                             SLOT(slotBlocksAttributes()));
+    contextMenu->addAction( tr("&Edit Block"), 0, actionHandler,
+                             SLOT(slotBlocksEdit()));
+    contextMenu->addAction( tr("&Insert Block"), 0, actionHandler,
+                             SLOT(slotBlocksInsert()));
+    contextMenu->addAction( tr("&Create New Block"), 0, actionHandler,
+                             SLOT(slotBlocksCreate()));
     contextMenu->exec(QCursor::pos());
     delete contextMenu;
 
@@ -455,20 +455,13 @@ void QG_BlockWidget::slotUpdateBlockList() {
         return;
     }
 
-    QRegExp rx("");
-    int pos = 0;
-    QString s, n;
-
-    n = matchBlockName->text();
-    rx.setPattern(n);
-    rx.setPatternSyntax(QRegExp::WildcardUnix);
+    QRegularExpression rx{ QRegularExpression::wildcardToRegularExpression(matchBlockName->text())};
 
     for (int i = 0; i < blockList->count(); i++) {
         RS_Block* block = blockModel->getBlock(i);
         if (!block) continue;
-        s = block->getName();
-        int f = rx.indexIn(s, pos);
-        if (!f) {
+        QRegularExpressionMatch match = rx.match(block->getName());
+        if (! match.hasMatch()) {
             blockView->showRow(i);
             blockModel->getBlock(i)->visibleInBlockList(true);
         } else {
diff --git a/librecad/src/ui/qg_graphicview.cpp b/librecad/src/ui/qg_graphicview.cpp
index 3db8c2ae..7367c811 100644
--- a/librecad/src/ui/qg_graphicview.cpp
+++ b/librecad/src/ui/qg_graphicview.cpp
@@ -33,6 +33,7 @@
 #include <QMenu>
 #include <QNativeGestureEvent>
 #include <QPoint>
+#include <QPointingDevice>
 #include <QTimer>
 
 #include "qc_applicationwindow.h"
@@ -606,7 +607,7 @@ bool QG_GraphicView::event(QEvent *event)
             }
 
             // It seems the NativeGestureEvent::pos() incorrectly reports global coordinates
-            QPoint g = mapFromGlobal(nge->globalPos());
+            QPoint g = mapFromGlobal(nge->globalPosition().toPoint());
             RS_Vector mouse = toGraph(g.x(), g.y());
             setCurrentAction(new RS_ActionZoomIn(*container, *this, direction,
 												 RS2::Both, &mouse, factor));
@@ -622,6 +623,52 @@ bool QG_GraphicView::event(QEvent *event)
  */
 void QG_GraphicView::tabletEvent(QTabletEvent* e) {
     if (testAttribute(Qt::WA_UnderMouse)) {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+        switch(e->pointerType()) {
+        case QPointingDevice::PointerType::Eraser:
+            if (e->type()==QEvent::TabletRelease) {
+                if (container) {
+
+                    RS_ActionSelectSingle* a =
+                        new RS_ActionSelectSingle(*container, *this);
+                    setCurrentAction(a);
+                    QMouseEvent ev(QEvent::MouseButtonRelease, e->position(), e->globalPosition(),
+                                   Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
+                    mouseReleaseEvent(&ev);
+                    a->finish();
+
+                    if (container->countSelected()>0) {
+                        setCurrentAction(
+                            new RS_ActionModifyDelete(*container, *this));
+                    }
+                }
+            }
+            break;
+
+        case QPointingDevice::PointerType::Generic:
+        case QPointingDevice::PointerType::Pen:
+        case QPointingDevice::PointerType::Cursor:
+            if (e->type()==QEvent::TabletPress) {
+                QMouseEvent ev(QEvent::MouseButtonPress, e->position(), e->globalPosition(),
+                               Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
+                mousePressEvent(&ev);
+            } else if (e->type()==QEvent::TabletRelease) {
+                QMouseEvent ev(QEvent::MouseButtonRelease, e->position(), e->globalPosition(),
+                               Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
+                mouseReleaseEvent(&ev);
+            } else if (e->type()==QEvent::TabletMove) {
+                QMouseEvent ev(QEvent::MouseMove, e->position(), e->globalPosition(),
+                               Qt::NoButton, {}, Qt::NoModifier);//RLZ
+                mouseMoveEvent(&ev);
+            }
+            break;
+        default:
+            break;
+
+
+        }
+
+#else
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
         switch (e->deviceType()) {
 #else
@@ -634,7 +681,7 @@ void QG_GraphicView::tabletEvent(QTabletEvent* e) {
                     RS_ActionSelectSingle* a =
                         new RS_ActionSelectSingle(*container, *this);
                     setCurrentAction(a);
-                    QMouseEvent ev(QEvent::MouseButtonRelease, e->pos(),
+                    QMouseEvent ev(QEvent::MouseButtonRelease, e->position(),
                                    Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
                     mouseReleaseEvent(&ev);
                     a->finish();
@@ -650,15 +697,15 @@ void QG_GraphicView::tabletEvent(QTabletEvent* e) {
         case QTabletEvent::Stylus:
         case QTabletEvent::Puck:
             if (e->type()==QEvent::TabletPress) {
-                QMouseEvent ev(QEvent::MouseButtonPress, e->pos(),
+                QMouseEvent ev(QEvent::MouseButtonPress, e->position(),
                                Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
                 mousePressEvent(&ev);
             } else if (e->type()==QEvent::TabletRelease) {
-                QMouseEvent ev(QEvent::MouseButtonRelease, e->pos(),
+                QMouseEvent ev(QEvent::MouseButtonRelease, e->position(),
                                Qt::LeftButton, Qt::LeftButton, Qt::NoModifier);//RLZ
                 mouseReleaseEvent(&ev);
             } else if (e->type()==QEvent::TabletMove) {
-                QMouseEvent ev(QEvent::MouseMove, e->pos(),
+                QMouseEvent ev(QEvent::MouseMove, e->position(),
                                Qt::NoButton, {}, Qt::NoModifier);//RLZ
                 mouseMoveEvent(&ev);
             }
@@ -667,6 +714,7 @@ void QG_GraphicView::tabletEvent(QTabletEvent* e) {
         default:
             break;
         }
+#endif
     }
 
     // a 'mouse' click:
@@ -698,7 +746,7 @@ void QG_GraphicView::leaveEvent(QEvent* e) {
     QWidget::leaveEvent(e);
 }
 
-void QG_GraphicView::enterEvent(QEvent* e) {
+void QG_GraphicView::enterEvent(QEnterEvent* e) {
     eventHandler->mouseEnterEvent();
     QWidget::enterEvent(e);
 }
@@ -877,6 +925,9 @@ void QG_GraphicView::wheelEvent(QWheelEvent *e) {
         QEvent::MouseMove,
 #if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
             e->position(),
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
+            e->globalPosition(),
+#endif
 #else
             QPointF{static_cast<qreal>(e->x()), static_cast<qreal>(e->y())},
 #endif
@@ -1226,7 +1277,11 @@ void QG_GraphicView::addScrollbars()
 
     setOffset(50, 50);
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    layout->setContentsMargins(QMargins{});
+#else
     layout->setMargin(0);
+#endif
     layout->setSpacing(0);
     layout->setColumnStretch(0, 1);
     layout->setColumnStretch(1, 0);
diff --git a/librecad/src/ui/qg_graphicview.h b/librecad/src/ui/qg_graphicview.h
index 3d3669be..30b1d7d0 100644
--- a/librecad/src/ui/qg_graphicview.h
+++ b/librecad/src/ui/qg_graphicview.h
@@ -36,7 +36,7 @@
 class QGridLayout;
 class QLabel;
 class QMenu;
-
+class QEnterEvent;
 class QG_ScrollBar;
 
 /**
@@ -110,7 +110,7 @@ protected:
 	void mouseMoveEvent(QMouseEvent* e) override;
 	void tabletEvent(QTabletEvent* e) override;
 	void leaveEvent(QEvent*) override;
-	void enterEvent(QEvent*) override;
+    void enterEvent(QEnterEvent*) override;
 	void focusInEvent(QFocusEvent*) override;
 	void focusOutEvent(QFocusEvent*) override;
 	void wheelEvent(QWheelEvent* e) override;
diff --git a/librecad/src/ui/qg_layerwidget.cpp b/librecad/src/ui/qg_layerwidget.cpp
index 17162af6..523afc30 100644
--- a/librecad/src/ui/qg_layerwidget.cpp
+++ b/librecad/src/ui/qg_layerwidget.cpp
@@ -159,7 +159,11 @@ QVariant QG_LayerModel::data ( const QModelIndex & index, int role ) const {
         }
         break;
 
+#if QT_VERSION > QT_VERSION_CHECK(6,0,0)
     case Qt::BackgroundRole:
+#else
+    case Qt::BackgroundColorRole:
+#endif
         if( COLOR_SAMPLE == col) {
             return layer->getPen().getColor().toQColor();
         }
@@ -516,18 +520,12 @@ void QG_LayerWidget::slotSelectionChanged(
  * Called when reg-expresion matchLayerName->text changed
  */
 void QG_LayerWidget::slotUpdateLayerList() {
-    QRegExp rx("");
-    int pos=0;
-    QString  s, n;
-
-    n=matchLayerName->text();
-    rx.setPattern(n);
-    rx.setPatternSyntax(QRegExp::WildcardUnix);
+    QRegularExpression rx = QRegularExpression::fromWildcard(matchLayerName->text());
 
     for (unsigned int i=0; i<layerList->count() ; i++) {
-        s=layerModel->getLayer(i)->getName();
-        int f=rx.indexIn(s, pos);
-        if ( !f ) {
+        QString s=layerModel->getLayer(i)->getName();
+        QRegularExpressionMatch match = rx.match(s);
+        if (match.hasMatch()) {
             layerView->showRow(i);
             layerModel->getLayer(i)->visibleInLayerList(true);
         } else {
diff --git a/librecad/src/ui/qg_librarywidget.cpp b/librecad/src/ui/qg_librarywidget.cpp
index 82a4e6be..f3eb12ae 100644
--- a/librecad/src/ui/qg_librarywidget.cpp
+++ b/librecad/src/ui/qg_librarywidget.cpp
@@ -34,6 +34,7 @@
 #include <QMouseEvent>
 #include <QPushButton>
 #include <QStandardItemModel>
+#include <QStandardPaths>
 #include <QTreeView>
 #include <QVBoxLayout>
 
@@ -388,7 +389,7 @@ QString QG_LibraryWidget::getPathToPixmap(const QString& dir,
         const QString& dxfPath) {
 
     // the thumbnail must be created in the user's home.
-    QString iconCacheLocation=QStandardPaths::writableLocation(QStandardPaths::DataLocation) + QDir::separator() + "iconCache" + QDir::separator();
+    QString iconCacheLocation=QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + QDir::separator() + "iconCache" + QDir::separator();
 
     RS_DEBUG->print("QG_LibraryWidget::getPathToPixmap: "
                     "dir: '%s' dxfFile: '%s' dxfPath: '%s'",
diff --git a/plugins/align/align.cpp b/plugins/align/align.cpp
index 235ff3c6..6a0d53ae 100644
--- a/plugins/align/align.cpp
+++ b/plugins/align/align.cpp
@@ -10,10 +10,10 @@
 /*  along with this program.  If not, see <http://www.gnu.org/licenses/>.    */
 /*****************************************************************************/
 
+#include <cmath>
 
 #include "document_interface.h"
 #include "align.h"
-#include <cmath>
 
 #include <QCheckBox>
 #include <QMessageBox>
diff --git a/plugins/align/align.pro b/plugins/align/align.pro
index 791aa30c..2f8dfc6d 100644
--- a/plugins/align/align.pro
+++ b/plugins/align/align.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.1
diff --git a/plugins/asciifile/asciifile.pro b/plugins/asciifile/asciifile.pro
index ee6be0b9..c91f9587 100644
--- a/plugins/asciifile/asciifile.pro
+++ b/plugins/asciifile/asciifile.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += gui widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.0
diff --git a/plugins/divide/divide.cpp b/plugins/divide/divide.cpp
index 4c4df6ae..4184a847 100644
--- a/plugins/divide/divide.cpp
+++ b/plugins/divide/divide.cpp
@@ -16,7 +16,6 @@
 
 #include <QtGui>
 #include <QApplication>
-#include <QDesktopWidget>
 #include <QDialog>
 #include <QGroupBox>
 #include <QDialogButtonBox>
diff --git a/plugins/divide/divide.pro b/plugins/divide/divide.pro
index 48201ac0..fb34f34a 100644
--- a/plugins/divide/divide.pro
+++ b/plugins/divide/divide.pro
@@ -1,4 +1,4 @@
-QT       += gui
+QT       += widgets
 TEMPLATE  = lib
 CONFIG   += plugin
 VERSION   = 0.0.1
diff --git a/plugins/divide/dividedlg.cpp b/plugins/divide/dividedlg.cpp
index 81776f00..6d406161 100644
--- a/plugins/divide/dividedlg.cpp
+++ b/plugins/divide/dividedlg.cpp
@@ -12,7 +12,6 @@
 
 #include <QtGui>
 #include <QApplication>
-#include <QDesktopWidget>
 #include <QDialog>
 #include <QGroupBox>
 #include <QDialogButtonBox>
diff --git a/plugins/list/list.cpp b/plugins/list/list.cpp
index 8fe8919c..271e50e4 100644
--- a/plugins/list/list.cpp
+++ b/plugins/list/list.cpp
@@ -12,7 +12,6 @@
 
 
 #include <QTextEdit>
-//#include <QColor>
 #include <QDialogButtonBox>
 #include <QVBoxLayout>
 #include <cmath>
diff --git a/plugins/list/list.pro b/plugins/list/list.pro
index 4d6294b9..6a09502f 100644
--- a/plugins/list/list.pro
+++ b/plugins/list/list.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.1
diff --git a/plugins/picfile/picfile.cpp b/plugins/picfile/picfile.cpp
index 790c74ae..4d4769eb 100644
--- a/plugins/picfile/picfile.cpp
+++ b/plugins/picfile/picfile.cpp
@@ -286,7 +286,7 @@ void picPunto::processFilePic(QFile* file)
                         QString txt = line.split("\"", skip).at(1);
                         QStringList rline = line.split("\"", skip).at(2).split(" ",skip);
                         txt.remove ( txt.size()-3, 3);
-                        txt.remove (QRegExp("^.*fR"));
+                        txt.remove (QRegularExpression("^.*fR"));
                         // printf("process line: %s: %s\n",rline.at(2).toStdString().c_str(),txt.toStdString().c_str() );
                         drawText(rline.at(2).split(',').at(0), rline.at(2).split(',').at(1), txt, rline.at(3));
                     }
diff --git a/plugins/picfile/picfile.pro b/plugins/picfile/picfile.pro
index 3d2de839..e7114ddf 100644
--- a/plugins/picfile/picfile.pro
+++ b/plugins/picfile/picfile.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.0
diff --git a/plugins/plotequation/plotequation.pro b/plugins/plotequation/plotequation.pro
index fc144950..6c380757 100644
--- a/plugins/plotequation/plotequation.pro
+++ b/plugins/plotequation/plotequation.pro
@@ -1,4 +1,4 @@
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.1
diff --git a/plugins/sameprop/sameprop.pro b/plugins/sameprop/sameprop.pro
index 93b8ca0a..adf9edc0 100644
--- a/plugins/sameprop/sameprop.pro
+++ b/plugins/sameprop/sameprop.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.1
diff --git a/plugins/sample/sample.pro b/plugins/sample/sample.pro
index 51880391..ce892d99 100644
--- a/plugins/sample/sample.pro
+++ b/plugins/sample/sample.pro
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += gui
+QT       += widgets
 TEMPLATE = lib
 CONFIG += plugin
 VERSION = 1.0.1
