commit 097bcc094a6fee93a61ff6400d9402af43cccd7b
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Sun Dec 17 02:32:08 2023 -0500

    Cal command: inline expression

diff --git a/librecad/src/cmd/rs_commands.cpp b/librecad/src/cmd/rs_commands.cpp
index 6ac93358..bc62f584 100755
--- a/librecad/src/cmd/rs_commands.cpp
+++ b/librecad/src/cmd/rs_commands.cpp
@@ -1380,7 +1380,7 @@ RS2::ActionType RS_Commands::keycodeToAction(const QString& code) const {
     if(!(code.startsWith(g_FnPrefix) ||
          code.startsWith(g_AltPrefix) ||
          code.startsWith(g_MetaPrefix))) {
-    	if(code.size() < 1 || code.contains(QRegExp("^[a-z].*",Qt::CaseInsensitive)) == false )
+    	if(code.size() < 1 || code.contains(QRegularExpression("^[a-z].*")) == false )
             return RS2::ActionNone;
         c = code.toLower();
     } else {
@@ -1363,7 +1363,7 @@ RS2::ActionType RS_Commands::keycodeToAction(const QString& code) const {
     }
 
 
-//    std::cout<<"regex: "<<qPrintable(c)<<" matches: "<< c.contains(QRegularExpression("^[a-z].*",Qt::CaseInsensitive))<<std::endl;
+    //    std::cout<<"regex: "<<qPrintable(c)<<" matches: "<< c.contains(QRegularExpression("^[a-z].*",Qt::CaseInsensitive))<<std::endl;
     //    std::cout<<"RS2::ActionType RS_Commands::keycodeToAction("<<qPrintable(c)<<")"<<std::endl;
 
     auto it = shortCommands.find(c);
@@ -1434,29 +1434,4 @@ QString RS_Commands::msgAvailableCommands() {
     return QObject::tr("Available commands:");
 }
 
-/**
- * @brief extractCliCal, filter cli calculator math expression
- * @param cmd, cli string
- * @return math expression for RS_Math:eval();
- */
-QString RS_Commands::filterCliCal(const QString& cmd)
-{
-
-    QString str=cmd.trimmed();
-    const QRegularExpression calCmd(R"(^(cal|calculate))");
-    if(!(str.contains(calCmd)
-         || str.startsWith(QObject::tr("cal","command to trigger cli calculator"), Qt::CaseInsensitive)
-         || str.startsWith(QObject::tr("calculate","command to trigger cli calculator"), Qt::CaseInsensitive)
-                           )) {
-        return QString();
-    }
-    int index=str.indexOf(QRegularExpression(R"(\s)"));
-    bool spaceFound=(index>=0);
-    str=str.mid(index);
-    index=str.indexOf(QRegularExpression(R"(\S)"));
-    if(!(spaceFound && index>=0)) return QString();
-    str=str.mid(index);
-    return str;
-}
-
 // EOF
diff --git a/librecad/src/ui/qg_commandedit.cpp b/librecad/src/ui/qg_commandedit.cpp
index db80ebf0..e28ed82a 100644
--- a/librecad/src/ui/qg_commandedit.cpp
+++ b/librecad/src/ui/qg_commandedit.cpp
@@ -34,9 +34,9 @@
 #include <QClipboard>
 
 #include "rs_commands.h"
-#include "rs_dialogfactory.h"
 #include "rs_math.h"
 #include "rs_settings.h"
+#include "rs_settings.h"
 
 
 /**
@@ -273,7 +273,7 @@ QString QG_CommandEdit::filterCliCal(const QString& cmd)
         bool okay=false;
         double result = RS_Math::eval(str, &okay);
         if (okay) {
-            RS_DIALOGFACTORY->commandMessage(QString("%1 =%2").arg(str).arg(result, 12, 'g'));
+            emit message(QString("%1 =%2").arg(str).arg(result, 12, 'g'));
             return {};
         }
     }
