commit 49168a02f97ff5438937c9d74b287ea8249ff907
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Wed Apr 10 23:26:57 2024 -0400

    qt6

diff --git a/librecad/src/lib/engine/rs_blocklist.cpp b/librecad/src/lib/engine/rs_blocklist.cpp
index a8b5c713..31f14dc4 100644
--- a/librecad/src/lib/engine/rs_blocklist.cpp
+++ b/librecad/src/lib/engine/rs_blocklist.cpp
@@ -241,8 +241,8 @@ QString RS_BlockList::newName(const QString& suggestion) {
 		return suggestion;
 
 	QString name=suggestion;
-	const QRegularExpression rx(R"(-\d+$)");
-	int index=name.lastIndexOf(rx);
+    QRegularExpression const rx(R"(-\d+$)");
+    int index=rx.match(name).lastCapturedIndex();
 	int i=-1;
 	if(index>0){
 		i=name.mid(index+1).toInt();
diff --git a/librecad/src/lib/engine/rs_hatch.cpp b/librecad/src/lib/engine/rs_hatch.cpp
index ec927cf8..1db34b8c 100644
--- a/librecad/src/lib/engine/rs_hatch.cpp
+++ b/librecad/src/lib/engine/rs_hatch.cpp
@@ -709,7 +709,7 @@ double RS_Hatch::getTotalArea() {
     try {
         getTotalAreaImpl();
     } catch(...) {
-        LC_ERR<<__func__<<"(): line "<<__LINE__<<": failure in finding hatch area";
+        RS_DEBUG->print(RS_Debug::D_ERROR, "RS_Hatch:: %s() failure in find hatch area", __func__);
     }
 
     return m_area;
diff --git a/librecad/src/lib/engine/rs_pattern.cpp b/librecad/src/lib/engine/rs_pattern.cpp
index dc84cb36..002a39b8 100644
--- a/librecad/src/lib/engine/rs_pattern.cpp
+++ b/librecad/src/lib/engine/rs_pattern.cpp
@@ -54,9 +54,7 @@ RS_Pattern::RS_Pattern(const QString& fileName)
  */
 RS_Entity* RS_Pattern::clone() const
 {
-    auto* cloned = new RS_Pattern(fileName);
-    *static_cast<RS_EntityContainer*>(cloned) = *static_cast<RS_EntityContainer*>(RS_EntityContainer::clone());
-    cloned->loaded = loaded;
+    auto* cloned = new RS_Pattern(*this);
     return cloned;
 }
 
diff --git a/librecad/src/lib/engine/rs_patternlist.cpp b/librecad/src/lib/engine/rs_patternlist.cpp
index f7cc1a17..e7ef5695 100644
--- a/librecad/src/lib/engine/rs_patternlist.cpp
+++ b/librecad/src/lib/engine/rs_patternlist.cpp
@@ -77,8 +77,13 @@ std::unique_ptr<RS_Pattern> RS_PatternList::requestPattern(const QString& name)
     if (patterns.count(name2) == 0 || patterns.at(name2) == nullptr) {
         auto p = std::make_unique<RS_Pattern>(name2);
         if (p!=nullptr) {
-            p->loadPattern();
-            patterns[name2].swap(p);
+            if (p->loadPattern()) {
+                patterns.emplace(name2,  std::unique_ptr<RS_Pattern>{});
+                patterns[name2].swap(p);
+            }
+            else {
+                patterns.erase(name2);
+            }
         }
         else {
             LC_ERR<<"RS_PatternList::"<<__func__<<"(): loading pattern failed: "<<name2;
diff --git a/librecad/src/ui/lc_penpalettewidget.cpp b/librecad/src/ui/lc_penpalettewidget.cpp
index b4da4e7d..238f7636 100644
--- a/librecad/src/ui/lc_penpalettewidget.cpp
+++ b/librecad/src/ui/lc_penpalettewidget.cpp
@@ -260,11 +260,13 @@ void LC_PenPaletteWidget::onTableViewContextMenuInvoked([[maybe_unused]] const Q
         caption->setAlignment(Qt::AlignCenter);
         typedef void (LC_PenPaletteWidget::*MemFn)();
         auto addAction = [&contextMenu, this](const std::pair<QString, MemFn>& item) {
-#if QT_VERSION >= QT_VERSION_CHECK(6, 2, 0)
-            contextMenu->addAction(item.first, {}, this, item.second);
-#else
-            contextMenu->addAction(item.first, this, item.second);
-#endif
+            auto* action = contextMenu->addAction(item.first);
+            connect(action, &QAction::triggered, this, item.second);
+// #if QT_VERSION >= QT_VERSION_CHECK(6, 2, 0)
+//             connect(action, &QAction::triggered, this, time.second);
+// #else
+//             connect(action, SIGNAL(triggered()), this, time.second);
+// #endif
         };
         auto addActions = [&addAction](std::initializer_list<std::pair<QString, MemFn>> menuEntries){
             for (const auto& menuEntry: menuEntries)
