diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index 1a35403..b006e92 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -141,7 +141,7 @@ def rust_compiler(rustc_info, cargo_info, build_project):
         rustc_min_version = Version('1.22.0')
         cargo_min_version = Version('0.23.0')
     else:
-        rustc_min_version = Version('1.34.0')
+        rustc_min_version = Version('1.32.0')
         cargo_min_version = rustc_min_version
 
     version = rustc_info.version
