diff --git a/third_party/rust/libloading/build.rs b/third_party/rust/libloading/build.rs
index 88c8c5f..268159f 100644
--- a/third_party/rust/libloading/build.rs
+++ b/third_party/rust/libloading/build.rs
@@ -13,7 +13,7 @@ fn main(){
         // found any libraries that have to be linked to on other platforms.
         // What happens if the executable is not linked up dynamically?
         Ok("openbsd") | Ok("bitrig") | Ok("netbsd") | Ok("macos") | Ok("ios") => {}
-        Ok("solaris") => {}
+        Ok("solaris") | Ok("illumos") => {}
         // dependencies come with winapi
         Ok("windows") => {}
         tos => {
