diff --git a/lang/c++/CMakeLists.txt b/lang/c++/CMakeLists.txt
index e8efe86..26f191a 100644
--- a/lang/c++/CMakeLists.txt
+++ b/lang/c++/CMakeLists.txt
@@ -133,6 +133,7 @@ enable_testing()
 
 macro (unittest name)
     add_executable (${name} test/${name}.cc)
+    target_compile_options (${name} PRIVATE -pthreads -D_PTHREADS)
     target_link_libraries (${name} avrocpp ${Boost_LIBRARIES})
     add_test (NAME ${name} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name})
diff --git a/lang/c++/CMakeLists.txt b/lang/c++/CMakeLists.txt
index 02ae881..12e6824 100644
--- a/lang/c++/CMakeLists.txt
+++ b/lang/c++/CMakeLists.txt
@@ -163,11 +163,11 @@ set (CPACK_PACKAGE_FILE_NAME "avrocpp-${AVRO_VERSION_MAJOR}")
 include (CPack)
 
 install (TARGETS avrocpp avrocpp_s
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
-    RUNTIME DESTINATION lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
-install (TARGETS avrogencpp RUNTIME DESTINATION bin)
+install (TARGETS avrogencpp RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 install (DIRECTORY api/ DESTINATION include/avro
     FILES_MATCHING PATTERN *.hh)
diff --git a/lang/c++/CMakeLists.txt b/lang/c++/CMakeLists.txt
index 41297e1..b11c6ec 100644
--- a/lang/c++/CMakeLists.txt
+++ b/lang/c++/CMakeLists.txt
@@ -59,6 +59,8 @@ add_definitions (${Boost_LIB_DIAGNOSTIC_DEFINITIONS})
 
 include_directories (api ${CMAKE_CURRENT_BINARY_DIR} ${Boost_INCLUDE_DIRS})
 
+find_library(STDC++_LIBRARY stdc++ PATHS /usr/gcc/6/lib/${LIB_SUFFIX}/ NO_DEFAULT_PATH )
+ 
 set (AVRO_SOURCE_FILES
         impl/Compiler.cc impl/Node.cc
         impl/NodeImpl.cc impl/ResolverSchema.cc impl/Schema.cc
@@ -129,6 +131,8 @@ macro (unittest name)
     add_executable (${name} test/${name}.cc)
     target_compile_options (${name} PRIVATE -pthread)
     target_link_libraries (${name} avrocpp ${Boost_LIBRARIES})
+    set_target_properties(${name} PROPERTIES 
+                      INSTALL_RPATH /usr/gcc/6/lib${LIB_SUFFIX})
     add_test (NAME ${name} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name})
 endmacro (unittest)
diff --git a/lang/c++/CMakeLists.txt b/lang/c++/CMakeLists.txt
index 8982889..b629038 100644
--- a/lang/c++/CMakeLists.txt
+++ b/lang/c++/CMakeLists.txt
@@ -48,7 +48,7 @@ add_definitions (
 endif()
 
 if (CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "-Wall")
+    set(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
 endif ()
 
 
