diff --git a/lang/c++/test/buffertest.cc b/lang/c++/test/buffertest.cc
index ac445aa..430b035 100644
--- a/lang/c++/test/buffertest.cc
+++ b/lang/c++/test/buffertest.cc
@@ -18,6 +18,10 @@
 
 #include <boost/test/included/unit_test_framework.hpp>
 
+#if BOOST_VERSION > 105300
+#define BOOST_MESSAGE(msg) BOOST_TEST_MESSAGE(msg)
+#endif
+
 #include <boost/thread.hpp>
 #include <boost/bind.hpp>
 #include <boost/scoped_array.hpp>
