diff --git a/src/bootstrap/install.rs b/src/bootstrap/install.rs
index f8734eb..bf7ae60 100644
--- a/src/bootstrap/install.rs
+++ b/src/bootstrap/install.rs
@@ -105,7 +105,7 @@ fn install_sh(
         pkgname(builder, name)
     };
 
-    let mut cmd = Command::new("sh");
+    let mut cmd = Command::new("bash");
     cmd.current_dir(&empty_dir)
         .arg(sanitize_sh(&tmpdir(builder).join(&package_name).join("install.sh")))
         .arg(format!("--prefix={}", sanitize_sh(&prefix)))
