diff --git a/nspr/configure b/nspr/configure
index 880d1a2..1bfb5a8 100755
--- a/nspr/configure
+++ b/nspr/configure
@@ -7302,12 +7302,10 @@ $as_echo "$as_me: WARNING: Unknown version of the Microsoft (R) Manifest Tool."
 
         fi
         CPU_ARCH_TAG=_$OS_TEST
-        # The default debug format, DWARF (-g), is not supported by gcc
-        # on i386-ANY-sysv4/solaris, but the stabs format is.  It is
+        # It is
         # assumed that the Solaris assembler /usr/ccs/bin/as is used.
         # If your gcc uses GNU as, you do not need the -Wa,-s option.
         if test -n "$MOZ_DEBUG" && test -n "$GNU_CC"; then
-            _DEBUG_FLAGS=-gstabs
             if test -z "$GNU_AS"; then
                 _DEBUG_FLAGS="$_DEBUG_FLAGS -Wa,-s"
             fi
