diff --git a/applets/clock/clock.c b/applets/clock/clock.c
index 3836e92..b46b121 100644
--- a/applets/clock/clock.c
+++ b/applets/clock/clock.c
@@ -51,10 +51,9 @@
 #include <gdk/gdkx.h>
 #include <gconf/gconf-client.h>
 
-#include <libgweather/gweather-prefs.h>
-#include <libgweather/gweather-xml.h>
-#include <libgweather/location-entry.h>
-#include <libgweather/timezone-menu.h>
+#include <libmateweather/mateweather-prefs.h>
+#include <libmateweather/location-entry.h>
+#include <libmateweather/timezone-menu.h>
 
 #ifdef HAVE_LIBECAL
 #include <libedataserverui/e-passwords.h>
@@ -137,8 +136,8 @@ struct _ClockData {
 	GtkWidget *prefs_location_edit_button;
 	GtkWidget *prefs_location_remove_button;
 
-	GWeatherLocationEntry *location_entry;
-        GWeatherTimezoneMenu *zone_combo;
+	MateWeatherLocationEntry *location_entry;
+        MateWeatherTimezoneMenu *zone_combo;
 
 	GtkWidget *time_settings_button;
 	GtkWidget *calendar;
diff --git a/applets/clock/clock.c b/applets/clock/clock.c
index a47b0fb..b7e9a15 100644
--- a/applets/clock/clock.c
+++ b/applets/clock/clock.c
@@ -2346,7 +2346,7 @@ parse_and_set_temperature_string (const char *str, ClockData *cd)
         gint value = 0;
 	gboolean use_default = FALSE;
 
-	value = gweather_prefs_parse_temperature (str, &use_default);
+	value = mateweather_prefs_parse_temperature (str, &use_default);
 
 	cd->use_temperature_default = use_default;
 	cd->temperature_unit = value;
@@ -2358,7 +2358,7 @@ parse_and_set_speed_string (const char *str, ClockData *cd)
         gint value = 0;
 	gboolean use_default = FALSE;
 
-	value = gweather_prefs_parse_speed (str, &use_default);
+	value = mateweather_prefs_parse_speed (str, &use_default);
 
 	cd->use_speed_default = use_default;
 	cd->speed_unit = value;
@@ -2945,11 +2945,11 @@ run_prefs_edit_save (GtkButton *button, ClockData *cd)
         const gchar *timezone, *weather_code;
         gchar *city, *name;
 
-        GWeatherLocation *gloc;
+        MateWeatherLocation *gloc;
         gfloat lat = 0;
         gfloat lon = 0;
 
-        timezone = gweather_timezone_menu_get_tzid (cd->zone_combo);
+        timezone = mateweather_timezone_menu_get_tzid (cd->zone_combo);
         if (!timezone) {
                 edit_hide (NULL, cd);
                 return;
@@ -2959,13 +2959,13 @@ run_prefs_edit_save (GtkButton *button, ClockData *cd)
         weather_code = NULL;
         name = NULL;
 
-        gloc = gweather_location_entry_get_location (cd->location_entry);
+        gloc = mateweather_location_entry_get_location (cd->location_entry);
         if (gloc) {
-                city = gweather_location_get_city_name (gloc);
-                weather_code = gweather_location_get_code (gloc);
+                city = mateweather_location_get_city_name (gloc);
+                weather_code = mateweather_location_get_code (gloc);
         }
 
-        if (gweather_location_entry_has_custom_text (cd->location_entry)) {
+        if (mateweather_location_entry_has_custom_text (cd->location_entry)) {
                 name = gtk_editable_get_chars (GTK_EDITABLE (cd->location_entry), 0, -1);
         }
 
@@ -3051,10 +3051,10 @@ static void
 fill_timezone_combo_from_location (ClockData *cd, ClockLocation *loc)
 {
         if (loc != NULL) {
-                gweather_timezone_menu_set_tzid (cd->zone_combo, 
+                mateweather_timezone_menu_set_tzid (cd->zone_combo, 
                                                  clock_location_get_timezone (loc));
         } else {
-                gweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
+                mateweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
         }
 }
 
@@ -3067,7 +3067,7 @@ location_update_ok_sensitivity (ClockData *cd)
 
         ok_button = _clock_get_widget (cd, "edit-location-ok-button");
 
-        timezone = gweather_timezone_menu_get_tzid (cd->zone_combo);
+        timezone = mateweather_timezone_menu_get_tzid (cd->zone_combo);
         name = gtk_editable_get_chars (GTK_EDITABLE (cd->location_entry), 0, -1);
 
         if (timezone && name && name[0] != '\0') {
@@ -3082,27 +3082,27 @@ location_update_ok_sensitivity (ClockData *cd)
 static void
 location_changed (GObject *object, GParamSpec *param, ClockData *cd)
 {
-        GWeatherLocationEntry *entry = GWEATHER_LOCATION_ENTRY (object);
-        GWeatherLocation *gloc;
-        GWeatherTimezone *zone;
+        MateWeatherLocationEntry *entry = MATEWEATHER_LOCATION_ENTRY (object);
+        MateWeatherLocation *gloc;
+        MateWeatherTimezone *zone;
         gboolean latlon_valid;
         double latitude = 0.0, longitude = 0.0;
 
-        gloc = gweather_location_entry_get_location (entry);
+        gloc = mateweather_location_entry_get_location (entry);
 
-	latlon_valid = gloc && gweather_location_has_coords (gloc);
+	latlon_valid = gloc && mateweather_location_has_coords (gloc);
         if (latlon_valid)
-                gweather_location_get_coords (gloc, &latitude, &longitude);
+                mateweather_location_get_coords (gloc, &latitude, &longitude);
         update_coords (cd, latlon_valid, latitude, longitude);
 
-        zone = gloc ? gweather_location_get_timezone (gloc) : NULL;
+        zone = gloc ? mateweather_location_get_timezone (gloc) : NULL;
         if (zone)
-                gweather_timezone_menu_set_tzid (cd->zone_combo, gweather_timezone_get_tzid (zone));
+                mateweather_timezone_menu_set_tzid (cd->zone_combo, mateweather_timezone_get_tzid (zone));
         else
-                gweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
+                mateweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
 
         if (gloc)
-                gweather_location_unref (gloc);
+                mateweather_location_unref (gloc);
 }
 
 static void
@@ -3126,8 +3126,8 @@ edit_clear (ClockData *cd)
         GtkWidget *lon_combo = _clock_get_widget (cd, "edit-location-longitude-combo");
 
         /* clear out the old data */
-        gweather_location_entry_set_location (cd->location_entry, NULL);
-        gweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
+        mateweather_location_entry_set_location (cd->location_entry, NULL);
+        mateweather_timezone_menu_set_tzid (cd->zone_combo, NULL);
 
         gtk_entry_set_text (GTK_ENTRY (lat_entry), "");
         gtk_entry_set_text (GTK_ENTRY (lon_entry), "");
@@ -3278,7 +3278,7 @@ edit_tree_row (GtkTreeModel *model, GtkTreePath *path, GtkTreeIter *iter, gpoint
 
         gtk_tree_model_get (model, iter, COL_CITY_LOC, &loc, -1);
 
-        gweather_location_entry_set_city (cd->location_entry,
+        mateweather_location_entry_set_city (cd->location_entry,
                                           clock_location_get_city (loc),
                                           clock_location_get_weather_code (loc));
 	name = clock_location_get_name (loc);
@@ -3363,7 +3363,7 @@ temperature_combo_changed (GtkComboBox *combo, ClockData *cd)
 	if (value == old_value)
 		return;
 
-	str = gweather_prefs_temp_enum_to_string (value);
+	str = mateweather_prefs_temp_enum_to_string (value);
 
 	panel_applet_gconf_set_string (PANEL_APPLET (cd->applet),
 				       KEY_TEMPERATURE_UNIT, str, NULL);
@@ -3386,7 +3386,7 @@ speed_combo_changed (GtkComboBox *combo, ClockData *cd)
 	if (value == old_value)
 		return;
 
-	str = gweather_prefs_speed_enum_to_string (value);
+	str = mateweather_prefs_speed_enum_to_string (value);
 
 	panel_applet_gconf_set_string (PANEL_APPLET (cd->applet),
 				       KEY_SPEED_UNIT, str, NULL);
@@ -3491,7 +3491,7 @@ fill_prefs_window (ClockData *cd)
 
         for (i = 0; temperatures[i] != -1; i++)
                 gtk_combo_box_append_text (GTK_COMBO_BOX (widget),
-                                           gweather_prefs_get_temp_display_name (temperatures[i]));
+                                           mateweather_prefs_get_temp_display_name (temperatures[i]));
 
 	update_temperature_combo (cd);
 	g_signal_connect (widget, "changed",
@@ -3507,7 +3507,7 @@ fill_prefs_window (ClockData *cd)
 
         for (i = 0; speeds[i] != -1; i++)
                 gtk_combo_box_append_text (GTK_COMBO_BOX (widget),
-                                           gweather_prefs_get_speed_display_name (speeds[i]));
+                                           mateweather_prefs_get_speed_display_name (speeds[i]));
 
 	update_speed_combo (cd);
 	g_signal_connect (widget, "changed",
@@ -3528,7 +3528,7 @@ ensure_prefs_window_is_created (ClockData *cd)
         GtkWidget *location_name_label;
         GtkWidget *timezone_label;
         GtkTreeSelection *selection;
-        GWeatherLocation *world;
+        MateWeatherLocation *world;
 
         if (cd->prefs_window) {
                 update_show_weather_toggles (cd);
@@ -3590,10 +3590,10 @@ ensure_prefs_window_is_created (ClockData *cd)
 
         edit_ok_button = _clock_get_widget (cd, "edit-location-ok-button");
 
-        world = gweather_location_new_world (FALSE);
+        world = mateweather_location_new_world (FALSE);
 
         location_box = _clock_get_widget (cd, "edit-location-name-box");
-        cd->location_entry = GWEATHER_LOCATION_ENTRY (gweather_location_entry_new (world));
+        cd->location_entry = MATEWEATHER_LOCATION_ENTRY (mateweather_location_entry_new (world));
         gtk_widget_show (GTK_WIDGET (cd->location_entry));
         gtk_container_add (GTK_CONTAINER (location_box), GTK_WIDGET (cd->location_entry));
         gtk_label_set_mnemonic_widget (GTK_LABEL (location_name_label),
@@ -3605,7 +3605,7 @@ ensure_prefs_window_is_created (ClockData *cd)
                           G_CALLBACK (location_name_changed), cd);
 
         zone_box = _clock_get_widget (cd, "edit-location-timezone-box");
-        cd->zone_combo = GWEATHER_TIMEZONE_MENU (gweather_timezone_menu_new (world));
+        cd->zone_combo = MATEWEATHER_TIMEZONE_MENU (mateweather_timezone_menu_new (world));
         gtk_widget_show (GTK_WIDGET (cd->zone_combo));
         gtk_container_add (GTK_CONTAINER (zone_box), GTK_WIDGET (cd->zone_combo));
         gtk_label_set_mnemonic_widget (GTK_LABEL (timezone_label),
@@ -3614,7 +3614,7 @@ ensure_prefs_window_is_created (ClockData *cd)
         g_signal_connect (G_OBJECT (cd->zone_combo), "notify::tzid",
                           G_CALLBACK (location_timezone_changed), cd);
 
-        gweather_location_unref (world);
+        mateweather_location_unref (world);
 
         g_signal_connect (G_OBJECT (edit_cancel_button), "clicked",
                           G_CALLBACK (edit_hide), cd);
